\documentclass{bioinfo}

\usepackage{xcolor}
\usepackage{colortbl}
\usepackage{cases}
\graphicspath{{./figures/}}
\definecolor{RubineRed}{RGB}{240, 0, 240}       % RubineRed  Approximate PANTONE RUBINE-RED
\usepackage{todonotes}


\copyrightyear{2016} \pubyear{2016}
\access{Advance Access Publication Date: Day Month Year}
\appnotes{Original Paper}


\begin{document}
\firstpage{1}

\subtitle{
\textcolor{red}{???Genetics and population analysis}\\
\textcolor{red}{???Genome analysis}
}

\title[Deconvolute mixed genomes]{DEploid: Untangling multiplicity of infection in {\it Plasmodium falciparum}.}
\author[Zhu \textit{et~al}.]{Sha Joe Zhu\,$^{\text{\sfb 1},*}$, Jacob Almagro Garcia\,$^{\text{\sfb 1}}$ and Gil McVean\,$^{\text{\sfb 1},*}$}
\address{$^{\text{\sf 1}}$Wellcome Trust Centre for Human Genetics, University of Oxford, Oxford OX3 7BN, UK }

\corresp{$^\ast$To whom correspondence should be addressed.}

\history{Received on \textcolor{red}{XXXXX}; revised on \textcolor{red}{XXXXX}; accepted on \textcolor{red}{XXXXX}}

\editor{Associate Editor: \textcolor{red}{XXXXXXX}}

\abstract{\textbf{Motivation:}
Multiplicity of infection in the malarial parasite {\it Plasmodium falciparum} affects  key phenotypic  traits, including drug resistance and risk of severe disease. Advances in protocols and sequencing technology have made possible  to obtain  high-coverage genome-wide  sequence data from blood samples taken in the field. However, analyzing and interpreting such data is challenging  because of the high rate of multiple  infections present in the field.\\
\textbf{Results:} The software package {\it DEploid} learns haplotype structure from a reference panel of  clonal isolates, and deconvolutes sequences of mixed samples. It reports the number of strains, the mixing proportions and the haplotypes present in an isolate. allowing researchers to study malaria infection history with an unprecedented level of detail.\\
\textbf{Availability and implementation:} The open source implementation {\it DEploid} is freely available at \href{https://github.com/mcveanlab/dEploid}{https://github.com/mcveanlab/dEploid} under the conditions of the GPLv3 license.\\
\textbf{Contact:} \href{joe.zhu@well.ox.ac.uk}{joe.zhu@well.ox.ac.uk} or \href{mcvean@well.ox.ac.uk}{mcvean@well.ox.ac.uk}\\
\textbf{Supplementary information:} Supplementary data are available at \textit{Bioinformatics} online.}

\maketitle

\section{Introduction}
Malaria is still one of the top global health problems. Transmitted by mosquitos of the genus Anopheles, the majority of malaria related deaths are caused by the  Plasmodium falciparum parasite (WHO, 2015). . Patients are often infected with  more than one parasite strain, due to bites from multiple mosquitoes, mosquitoes carrying multiple genetic types or a combination of both. Multiplicity of infection  can lead to competitions among co-existing strains  and may increase  disease development \citep{deRoode2005}, higher transmission rates \citep{Arnot1998} and even the spread of drug resistance \citep{deRoode2004}.

The presence of multiple strains of {\it P. falciparum} makes fine scale analysis of genetic variation very challenging since genetic differences between the genetic types of this haploid organism will render as heterozygous loci. Mixed calls also confounds methods that exploit haplotype data to detect, among other phenomena, the occurrence of natural selection or recent demographic events. In light of these difficulties, researchers usually focus on clonal infections or resort to heuristics methods for resolving heterozygous genotypes. The former approach discards valuable information regarding genetic diversity and inbreeding whereas the latter tends to create chimeric haplotypes that are not suitable for analysis, unless mixed calls are very sparse.

{\it Phasing} or deconvoluting the strains of a mixed infection is a harder problem than phasing diploid organisms because the levels of mixture within isolates (i.e. the abundance of each genetic type) vary greatly and are unknown.  Existing tools for phasing diploid organisms, such as Beagle \citep{Browning2007} and IMPUTE2 \citep{Howie2009}, are not designed to cope with this. \citet{Galinsky2015} and \citet{Jack2016} have attempted to address the mixed infection problem by solving the mixed proportion from allele frequencies, yet the haplotypes within a mixed isolate remain unclear.


As part of the Pf3k project \citep{Pf3k2016}, an effort to map the genetic diversity of {\it P. falciparum} at global scale, we have developed the {\tt DEploid}, a software package for deconvoluting  mixed infections. The program provides estimates for the number of different genetic types present in the isolate, the proportion or abundance of each strain and their sequences (i.e. haplotypes). To our knowledge, DEploid is the first package able to deconvolute strain haplotypes and provides a unique opportunity for researchers to study inbreeding and infection history at fine scale.


\begin{methods}

\section{Methods}
Overall, we use Markov chain Monte Carlo (MCMC) methods to learn the number of parasite strains and the proportions of allele frequencies, and use sampling method to infer the haplotype of each strain.

%In practice, more parasite strains are assumed than it needs to be. Thus, the number of parasite strains are fixed; and the noisy strains with proportions below one percent are ``zeroed-out''. More specifically,
%\begin{itemize}
%\item we take into account over-dispersion in read counts, modelling the read count distribution as a Beta-binomial distribution: The read counts of alternative alleles are identically and independently distributed (i.i.d.) with a probability, which is a random variable with a beta-distribution.

%\item A Metropolis-Hastings algorithm is used to sample proportions given the haplotypes, and updated sparsely.
%\end{itemize}

%We treat the clonal sample sequences as the back-bone of the haplotype structure, combine with information from the targeted sequence data to infer haplotypes by sampling. Specifically, we use \citet{Li2003}â€™s hidden Markov model framework as a starting point, with the following modifications:

\subsection{Notations}
Let's first introduce some of the notation. Suppose that our data $D$ are the allele counts of sample $j$ at a given site $i$, denoted as $r_{j,i}$ and $a_{j,i}$ for reference and alternative allels respectively. The allele frequencies within sample (WSAF) $p_{j,i}$ and at the population level (PLAF) $f_i$ can be calculated by $ \frac{a_{j,i}}{a_{j,i} + r_{j,i}}$ and $ \frac{\sum_j a_{j,i}}{\sum_j a_{j,i} + \sum_j r_{j,i}}$.

Since all data in this section is subjected to the same sample, we drop the subscript $j$ from now on. Let $\mathbf w = [w_1,\dots, w_k]$ and $\mathbf{h}_{i} = [h_{1,i},\dots,h_{k,i}]$ denote the proportions and haplotypes of $k$ parasite strain at site $i$. \citet{Jack2016} suggest to express the expected WSAF $q_{i}$:
\begin{equation}
q_i= (\mathbf{w}\cdot\mathbf{h}_{i})  =  \sum_{k=1}^{K} w_k \cdot h_{k,i} .\label{eqn:qij_full_sum}
\end{equation}.

\subsubsection{Likelihood of data given the expected WSAF}
Suppose unadjusted allele frequency is $q_i$, given the reads error rate $e$, the expected allele frequency of `REF' read as `ALT' is $(1 - q_i)e$, and the expected allele frequency of `ALT' read as `REF' is $q_ie$. Thus, we adjust the WSAF take into account of read error as follows:
\begin{equation}
\pi_i = q_i + (1 - q_i)e - q_ie = q_i + (1 - 2q_i)e.\label{eqn:adj_q}
\end{equation}

We take into account over-dispersion in read counts, modelling the count distribution as a Beta-binomial distribution. Specifically, the read counts of `ALT' are identically and independently distributed (i.i.d.) with probability $\pi_{i}$ (adjusted), i.e. $a_i \sim Binom(\pi_{i}, a_i + r_i)$, and $\pi_{i}\sim Beta(\alpha, \beta)$, where $\pi_{i} = \alpha/(\alpha+\beta)$. From exprience, we set $\alpha = 100\cdot q_{i} $ and $\beta = 100\cdot (1-q_{i})$. Hence we can express the likelihood of the data using:
\begin{equation}
L(q_{i} | D) = P(D|q_{i}) \propto \frac{\Gamma(a_i + 100\cdot \pi_{i}) \Gamma(r_i + 100\cdot (1-\pi_{i}))}{\Gamma(100\cdot \pi_{i})\Gamma(100\cdot (1-\pi_{i}))}, \label{eqn:llk}
\end{equation}
of which expected WSAF $q_i$ is adjusted through Eqn.\eqref{eqn:adj_q}.

%\begin{table}[h]\centering
%\begin{tabular}{c|c}\hline
%Marker index & $i$ \\
%Sample index & $j$ \\
%Read count for reference allele & $r$ \\
%Read count for alternative allele & $a$ \\
%Population level allele frequency (PLAF) & $f$ \\
%Number of strains within sample & $k$ \\
%Proportion of strains & $\mathbf w$ \\
%haplotypes of $k$ parasite strain at site $i$ & $\mathbf{h}_{i}$ \\
%Observed within sample allele frequency (WSAF) & $p$ \\
%Unadjusted expected WSAF & $q$ \\
%Adjusted expected WSAF & $\pi$ \\\hline
%\end{tabular}
%\caption{Notation summary}
%\end{table}

\begin{table}[h]\centering
\begin{tabular}{c|c}\hline
$i$              & Marker index\\
$j$              & Sample index \\
$r$              & Read count for reference allele \\
$a$              & Read count for alternative allele \\
$f$              & Population level allele frequency (PLAF) \\
$k$              & Number of strains within sample \\
$\mathbf w$      & Proportion of strains \\
$\mathbf{h}_{i}$ & haplotypes of $k$ parasite strain at site $i$ \\
$p$              & Observed within sample allele frequency (WSAF) \\
$q$              & Unadjusted expected WSAF  \\
$\pi$            & Adjusted expected WSAF \\\hline
\end{tabular}
\caption{Notation summary}
\end{table}

\subsection{Technical details}
Overall, we generate MCMC samples for the proportions $\mathbf w$ and the haplotypes $\mathbf h$ for given number of strains. In particular, we assume there are more strains than we actually need, start the MCMC chain with a fixed $k$. As the values of proportion drops, ``zero-out'' the ``noisy'' strain. As for the MCMC moves, we use a Metropolis-Hastings algorithm to sample proportions $\mathbf w$ given $\mathbf h$ (see section~\ref{sec:updateP}); and use Gib sampler to update $\mathbf h$ of given $\mathbf w$, which are further divied into cases when building the reference panel (see section~\ref{sec:ref:build}) and deconvolute mixed samples (see section~\ref{sec:deconvolute}).

\subsubsection{MCMC update for proportions}\label{sec:updateP}
We use a sparse update on $\mathbf w$. We introduce a multivariate normal variable titre ${\mathbf x} = [x_1,\dots,x_k]$, where each $x$ is i.i.d. normally distributed from $N(0, 3)$, with the density function $d(x)$. We sample $x$s, then transform to $w = e^x$. We then normalise vector $\mathbf w$ by the sum, to obtain a new sample ${\mathbf w}$. Thus, the density of $\mathbf p$ is equivlent to the product of $d(x)$s, which leads us to
The prior ratio is equal to $\frac{\prod_i^k d(x'_k)}{\prod_i^k d(x'_k)}$; and the Hastings ratio is 1. Note that the move from $x$ to $x'$, $\delta x$ is symmetrical.

%The key step of the Metropolis Hastings algorithm is to compute the acceptance ratio, which is $$\frac{L(\mathbf{p'}|D) P(\mathbf{p'}) / \phi(\mathbf{p'}|\mathbf{p})}{L(\mathbf{p}|D) P(\mathbf{p}) / \phi(\mathbf{p}|\mathbf{p'})}.$$

%Let $\frac{L(\mathbf{p'}|D)}{L(\mathbf{p}|D)}$ be the likelihood ratio, which is $\exp( l(\mathbf{w'}|D) - (\mathbf{w}|D))$.


\subsection{Infer the reference strains}\label{sec:ref:build}
In practice, we assume clonal sample haplotypes capture the diversities of haplotype structures given all samples. We use them as the reference strains for start, and deconvolute the rest mixed samples from them. We start with a set of clonal sample candidate, and run the algorithm to confirm they are in fact clonal. We use Gib sampler to update $\mathbf h$ of given $\mathbf w$, randomly select one strain at the time, or a pair of strains to update in order to improve the mixing of the MCMC process.

\subsubsection{Update a single strain at one time}

Choose haplotype strain $s$ uniformly at random from these $K$ strains, consider both cases of updating the state of strain $s$ at position $i$ to $0$ and $1$, we compute the WSAF and its associated likelihood as follows: First of all, regardless what state that strain $s$ at position $i$ has, we need to remove it from the current WSAF, i.e. subtract $ w_s \cdot h_s$ from Eqn.~(\ref{eqn:qij_full_sum}), which gives
\begin{equation}
q_{i,-s} = \sum_{k\neq s} w_k \cdot h_k = \textrm{Eqn.~\eqref{eqn:qij_full_sum}} -  w_s \cdot h_s \label{eqn:qij_full_sum_minus_s}
\end{equation}

Therefore, updating strain $s$ of state $0$ and $1$, so the WSAF becomes
\begin{align}
q_{i,g_s=0} & = \textrm{Eqn.~\eqref{eqn:qij_full_sum_minus_s}} \label{eqn:qij0}\\
q_{i,g_s=1} & = \textrm{Eqn.~\eqref{eqn:qij_full_sum_minus_s}} + w_s \times 1 \label{eqn:qij1}
\end{align}
Substitude equations~\eqref{eqn:qij0} and \eqref{eqn:qij1} into Eqn.~\eqref{eqn:llk} to compute associated likelihood $L(q_{i,g_s=*} |D)$, which is expressed as $L(g_s=* |D)$ in short, for the rest of the paper.


%\subsubsection{Update haplotypes without LD}\label{sec:update_single_no_LD}
As one of our MCMC step to update the haplotypes, we sample the state (genotype) of strain $s$ at each postion according to the posterior probability at site $i$,
\begin{equation}
P(g_s = * | D) \propto L(g_s=*|D) \times P(g_s = *).\label{eqn:post:LDfree}
\end{equation}
%where $P(g_s = 1) = f_i$ and $P(g_s = 0) = 1-f_i$ when haplotype structure is unknown. Given the state of the sample $*$, we update the likelihood $L(g_s = *|D)$ using equations \eqref{eqn:qij0} and \eqref{eqn:qij1}.

%The conditional probabilities of genotype of the $s$th strain is $0$: \\
%$P(s = 0 | D) \propto P(s = 0)\times L(q_{i,j,s=0}|D)$, where $P(s = 0)$ is the prior probability. Assume no haplotype structure, $P(s = 0) = 1-PLAF$. Similarly, $P(s = 1 | D) \propto P(s = 1)\times L(q_{i,j,s=1}|D)$ and when no haplotype structure is assumed $P(s = 1) = PLAF$.

%Then update the likelihood at each position according to the sample.


%\subsubsection{HMM forward algorithm }

%Suppose the hidden states are $X_i$s and observations as $Y_i$s, and transition probabily as $p(x_{i} | x_{i-1})$, and the emission probability is $p(y_{i}|x_i)$. According to the forward algorithm, To demonstrate the recursion, let
%$$\alpha_t(x_t) = p(x_t,y_{1:t}) = \sum_{x_{t-1}}p(x_t,x_{t-1},y_{1:t}).$$
%Using the chain rule to expand $p(x_t,x_{t-1},y_{1:t})$, we can then write

%$$\alpha_t(x_t) = \sum_{x_{t-1}}p(y_t|x_t,x_{t-1},y_{1:t-1})p(x_t|x_{t-1},y_{1:t-1})p(x_{t-1},y_{1:t-1}).$$
%Because $y_t$ is conditionally independent of everything but $x_t$, and $x_t$ is conditionally independent of everything but $x_{t-1}$, this simplifies to

%\begin{equation}
%\alpha_t(x_t) = p(y_t|x_t)\sum_{x_{t-1}}p(x_t|x_{t-1})\alpha_{t-1}(x_{t-1}).
%\end{equation}


\subsubsection{Update two haplotypes at one time}
In order to improve the MCMC mixing, we update two haplotypes at one time. Suppose random sampling two strains to update, namely, $s_1$ and $s_2$. Similar to Eqn.~\eqref{eqn:qij_full_sum_minus_s}, we have
\begin{equation}
\begin{split}
q_{i,-s_1, -s_2} & ~ = ~ \sum_{k\neq s_1,s_2} w_k \cdot h_k \\
                 & ~ = ~ \textrm{Eqn.~\eqref{eqn:qij_full_sum}} - w_{s_1} \cdot h_{s_1} - w_{s_2} \cdot h_{s_2} \label{eqn:qij_full_sum_minus_s1_s2}
\end{split}
\end{equation}
Further more, we have
\begin{align}
q_{i,g_{s_1}=0,g_{s_2}=0} & = \textrm{Eqn.~\eqref{eqn:qij_full_sum_minus_s1_s2}} \label{eqn:qij00}\\
q_{i,g_{s_1}=0,g_{s_2}=0} & = \textrm{Eqn.~\eqref{eqn:qij_full_sum_minus_s1_s2}} + \cdot w_{s_1} \times 1 \label{eqn:qij10}\\
q_{i,g_{s_1}=0,g_{s_2}=1} & = \textrm{Eqn.~\eqref{eqn:qij_full_sum_minus_s1_s2}} + \cdot w_{s_2} \times 1 \label{eqn:qij01}\\
q_{i,g_{s_1}=0,g_{s_2}=1} & = \textrm{Eqn.~\eqref{eqn:qij_full_sum_minus_s1_s2}} + \cdot w_{s_1} \times 1 + w_{s_2} \times 1 \label{eqn:qij11}
\end{align}
Substitue expressions.~\eqref{eqn:qij00} to~\eqref{eqn:qij11}, into Eqn.~\eqref{eqn:llk}, we then obtain their associated likelihood $L(q_{i,g_{s_1}=*,g_{s_2}=\#} \big{|} D)$, which is denoted as $L(g_{s_1}=*,g_{s_2}=\# \big{|} D)$ in the rest of the paper.
%\begin{align}
%L_{00} = L(g_{s_1} = 0, g_{s_2} = 0|Data) \label{eqn:l_00_update_two_hap}\\
%L_{10} = L(g_{s_1} = 1, g_{s_2} = 0|Data) \label{eqn:l_10_update_two_hap}\\
%L_{01} = L(g_{s_1} = 0, g_{s_2} = 1|Data) \label{eqn:l_01_update_two_hap}\\
%L_{11} = L(g_{s_1} = 1, g_{s_2} = 1|Data) \label{eqn:l_11_update_two_hap}\\
%\end{align}

Similar to Eqn.~\eqref{eqn:post:LDfree}, we sample the state (genotype) of strains $s_1$ and $s_2$ simultaneously at each postion according to the posterior probability at site $i$:
\begin{equation}
\begin{split}
P(g_{s_1}=*,g_{s_2}=\#  | D) \propto & L(g_{s_1}=*,g_{s_2}=\# |D) \times \\
                                     & P(g_{s_1}=*,g_{s_2}=\# ),\label{eqn:post.two:LDfree}
\end{split}
\end{equation}
where $P(g_{s_1}=*,g_{s_2}=\# ) = P(g_{s_1}=*) \cdot P(g_{s_2}=\# )$, assume $s_1$ and $s_2$ are independent. % In particular, $P(g_s = 1) = f_i$ and $P(g_s = 0) = 1-f_i$. %Given the states of $s_1$ and $s_2$ , we update the likelihood $L(g_{s_1}=*,g_{s_2}=\# |D)$ using equations from \eqref{eqn:qij00} to \eqref{eqn:qij11}.




\subsection{Deconvolute the mixed isolates}

We use \citet{Li2003}'s hidden Markov model frame work as a starting point. The following modifications are made:
\begin{itemize}
\item likelihood of data given the expected WSAF rather than the ``product of approximate conditionals'' (PAC).
\item multiple strais with variable proportion rather than two sequences with equal probability.
\item simplifying the mutation model with a fixed miss copying operation.
\end{itemize}

\subsubsection{Update single haplotype with LD}

\paragraph{Recombination map model}

The first case refers to staying on the same path and the second to a recombination event (i.e switch).
Let $\psi_i$ is given by $\psi_i = N_e G_i$, with $N_e$ being the effective population size and $G_i$ the genetic distance between loci $i$ and $i+1$. We assume a uniform recombination map, genetic distances are computed by
$G_i = D_i / morgan$
where $D_i$ denotes the physical distance between loci $i$ and $i+1$ in nucleotide, $morgan$ is the average morgan distance, which we use 1500000, $N_e = 10$.

Whereas recombination probabilities for a segment are computed by the following function. Note that {\bf we scale the probabilities with the number of haplotypes in the reference panel}. Let $\textrm{RP}$ denote the set of the strains in the reference panel. For position $i > 1$, let $\rho_i'$ denote the probability of {\bf no} recombinations from site $i-1$ to $i$, we have $\rho'_i = \exp(-\psi_i)$. Thus, the probability of recombining from any strain in the panel is $\displaystyle\frac{1-\rho_i'}{|RP|}$, where $|RP|$ is the size of the panel.


A crucial difference between our method and \citet{Li2003}'s model is that mixed samples can have more than two strains, with unknown proportions. We randomly choose the strains to update, then apply LS's algorithm to sample the path using Gibbs sampler given the proportion $\mathbf p$ (see example in Fig.~\ref{fig:ls}.

\begin{figure}[ht]
\centering
\includegraphics[width=0.5\textwidth]{coupled-painting.png}
\caption{Illustration of \citet{Li2003}'s algorithm. Strain 1 haplotype is made up from reference haplotype segments of 1 and 2; and strain 2 haplotype is made up from reference haplotype segments of 3 and 4. With miss copying, we allow strain states differ from the path: At the third last position of strain 1, the path is copied from reference haplotype 2, with the state of ``0''.
}\label{fig:ls}
\end{figure}

In addiontion to updating the haplotypes from the panel, we take into account of miss copying (see example shown in Fig.~\ref{fig:ls}), which allow the actual genotype differ from the path, in order to improve the likelihood of data.

\begin{enumerate}
\item Consider the likelihood as the emission probabilities at site $i$. Let's use $g_p$ and $g_s$ to denote the genotype of the copied path and the updated strain respectively. We have:
\begin{equation}
\begin{split}
L(g_p = * | D) ~ = ~ & L(g_s = * \big{|} D) \times P(g_p = g_s) + \\
                     & L(g_s = 1 - * \big{|} D) \times P(g_p \neq g_s) \label{eqn:gp_given_D}
\end{split}
\end{equation}
where $*\in \{0,1\}$, and $1-*$ indicates the event that $g_s$ takes value that differs from $g_p$. Let $\mu$ denote the probability of miss copying, we have
$$\begin{cases}
P(g_p = g_s) &= 1-\mu, \\
P(g_p \neq g_s) &= \mu .\end{cases}$$

\item Compute the probability of path at each position using forward algorithm. Therefore, we have the posterior probabilility of path (reference strain) $p$ at position $i$ as:
\begin{equation}
\begin{split}
P_i(g_p \big{|} D) ~ \propto ~ & L(g_p \big{|} D) ~ \times \\
                               & \left( \rho_i' \cdot P_{i-1}(g_p \big{|} D)  +  \frac{1-\rho_i'}{|RP|} \cdot \sum_{x\in R} P_{i-1} (g_x \big{|} D) \right).
\end{split}\label{eqn:post_path}
\end{equation}
In the HMM frame work, $L(g_p \big{|} D)$ is the emmission probability of oberving data $D$ given the hidden state of the path, $\rho_i'$ and $\frac{1-\rho_i'}{|RP|}$ are the transition probabilities from position $i-1$ to $i$, of which reflect the recombination event in our context.

\item Sample the path up to position $i$, i.e. backwards, start from the end of the sequence. At the end position, sample path according $f_{u,end}$.
for the $i-1$ position, first sample if a recombination events had happened with the probabilities proportional to
$$
\begin{cases}
\rho_i' \cdot f_{u,i-1} & \text{no recombined},\\
\displaystyle ( 1-\rho_i' ) \cdot \sum_{x\in R} f_{x,i-1} & \text{recombined}.
\end{cases}
$$
If it was recombined, sample the path $u$, according to $f_{u,i-1}$.

\item Ultermately, given the state of the path at each site, we now want to sample the genotype according to the posterior probabilities:
\begin{equation}
P(g_{s} = * \big{|} D) =
\begin{cases}
P(g_{p} = * \big{|} D) \cdot (1-\mu), & g_s = g_p;\\
P(g_{p} = 1 - * \big{|} D) \cdot \mu, & g_s \neq g_p.
\end{cases}
\label{eqn:ps0}
\end{equation}
%which is equivlent to
%$$
%P(g_{s} = * \big{|} D) =
%\begin{cases}
%f_{u,i-1} \cdot (1-\mu) & g_{s} = *, g_{u} = *, \\
%f_{u,i-1} \cdot (\mu) & g_{s} = *, g_{u} = 1 - *.
%\end{cases}
%%%& P(g_{s} = *, g_{u} = * \big{|} D) + P(g_{s} = *, g_{u} = 1 - * \big{|} D) \label{eqn:ps0}\\
                 %%%= & (1 - \mu) \times P(g_{u} = * \big{|} D) + \mu \times P(g_{u} = 1 - * \big{|} D) ;
%$$


%\begin{equation}
%\begin{split}
%P(g_{s,i} = 1 | Data) = & P(g_{s,i} = 1, g_{path,i} = 0| Data) + P(g_{s,i} = 1, g_{path,i} = 1 | Data) \label{eqn:ps1}\\
                      %= & P(g_{s,i} = 1 | g_{path,i} = 0) \times P(g_{path,i} = 0 | Data) + \\
                        %& P(g_{s,i} = 1 | g_{path,i} = 1) \times P(g_{path,i} = 1 | Data) .
%\end{split}
%\end{equation}
%Since,
%\begin{align*}
%p.miss.copy = & P( g_{s,i} = 0 | g_{path,i} = 1) = P( g_{s,i} = 0 | g_{path,i} = 1), \\
%1-p.miss.copy = & P( g_{s,i} = 1 | g_{path,i} = 1) = P( g_{s,i} = 0 | g_{path,i} = 0).
%\end{align*}
%Alternatively, given the state of the path, we can sample the genotype of according to the following probabilities, by rearranging Eqn~\eqref{eqn:ps0} and \eqref{eqn:ps1}:
%and we have

%$$
%P(g_{s,i} | Data) =
%\begin{cases}
%P(g_{path,i} | Data) \times p.miss.copy + (1-P(g_{path,i} | Data)) \times (1-p.miss.copy) & g_{s,i} \neq g_{path,i},\\
%P(g_{path,i} | Data) \times (1-p.miss.copy) + (1-P(g_{path,i} | Data)) \times p.miss.copy & g_{s,i} = g_{path,i}.
%\end{cases}
%$$
\end{enumerate}


\subsubsection{Update pair of haplotypes with LD}\label{sec:deconvolute}
Similarly to the previous section, we need to
\begin{enumerate}
\item Compute the emission probabilities

\begin{equation}
\begin{split}
L(g_{p_1} = *, g_{p_2} = \# \big{|} D) = & P(g_{p_1} = g_{s_1}, g_{p_2} = g_{s_2}) \cdot L(g_{s_1} = *, g_{s_2} = \# \big{|} D) + \\
                                         & P(g_{p_1} = g_{s_1}, g_{p_2} \neq g_{s_2}) \cdot L(g_{s_1} = *, g_{s_2} = 1-\# \big{|} D) + \\
                                         & P(g_{p_1} \neq g_{s_1}, g_{p_2} = g_{s_2}) \cdot L(g_{s_1} = 1-*, g_{s_2} = \# \big{|} D) + \\
                                         & P(g_{p_1} \neq g_{s_1}, g_{p_2} \neq g_{s_2}) \cdot L(g_{s_1} = 1-*, g_{s_2} = 1-\# \big{|} D)
\end{split}\label{eqn:gp_given_D:two}
\end{equation}
where
\begin{align*}
P(g_{p_1} = g_{s_1}, g_{p_2} = g_{s_2})       & = (1-\mu)\cdot(1-\mu) , \\
P(g_{p_1} \neq g_{s_1}, g_{p_2} = g_{s_2})    & = \mu\cdot(1-\mu),\\
P(g_{p_1} = g_{s_1}, g_{p_2} \neq g_{s_2})    & = \mu\cdot(1-\mu),\\
P(g_{p_1} \neq g_{s_1}, g_{p_2} \neq g_{s_2}) & = \mu \cdot \mu.
\end{align*}


\item
Compute the probability of path at each position using forward algorithm.
%Let $R$ denote the set of the strains in the reference panel.
%$$f_{u,i|i-1}=
%\begin{cases}
%p.no.recomb   & path_i = path_{i-1}, \\
%\frac{1-p.no.recomb}{|R|} & path_i =*_{i-1}, \forall *\in R.
%\end{cases}
%$$
%Now, we need to consider pair of path, so the probabilities can be
%\begin{align*}
%p.no.recomb &* p.no.recomb\\
%p.no.recomb &*\frac{1-\rho_i'}{|RP|}\\
%\frac{1-p.no.recomb}{|R|} &* \frac{1-\rho_i'}{|RP|} \\
%\end{align*}
%Therefore, we have the probabilility of the $u$th and $v$th reference strains at position $i$ as:

Similar to Equation~\eqref{eqn:post_path}, for all possible pair of the copying strain, we take into account of the possiblility of one strain recombines and the other does not with the probability of $\rho_i' \cdot \frac{1-\rho_i'}{|RP|}$; both recombines, with the probability of $\rho_i' \cdot \rho_i'$; neither recombines, with the probability of $\frac{1-\rho_i'}{|RP|} \cdot\frac{1-\rho_i'}{|RP|}$, assuming that recombination events of two copying strains are independent from each other.
\begin{equation}
\begin{split}
P_{i}(g_{p_1},g_{p_2}\big{|}D) \propto \left[\right. & P_{i-1}(g_{p_1},g_{p_2}\big{|}D) \cdot \rho_i' \cdot \rho_i' + \\
                                         & \sum_{x\in R} P_{i-1}(g_{p_1},g_{x}\big{|}D) \cdot \rho_i' \cdot \frac{1-\rho_i'}{|RP|} + \\
                                         & \sum_{y\in R} P_{i-1}(g_{y},g_{p_2}\big{|}D) \cdot \rho_i' \cdot\frac{1-\rho_i'}{|RP|}+ \\
                                         & \sum_{x,y\in R\cdot R} P_{i-1}(g_{x},g_{y}\big{|}D)  \cdot\frac{1-\rho_i'}{|RP|} \cdot\frac{1-\rho_i'}{|RP|} \left.\right] \cdot L(g_{p_1},g_{p_2} \big{|} D) \label{eqn:prob.update.two}
\end{split}
\end{equation}

\item
Sample the path up to position $i$, i.e. backwards, start from the end of the panel. At the end position, sample path according $P(p_1 = u, p_2 =v) = f_{u,v,end}$.
for the $i-1$ position, first sample if a recombination events had happened given the probabilities of
\begin{numcases}
\\
f_{u,v,i-1} \cdot \rho_i' \cdot \rho_i', & \text{no recombined},\\
\sum_{*\in RP}f_{u,*,i-1} \cdot \frac{1-\rho_i'}{|RP|} \cdot \rho_i', & u \text{ recombined} \label{eqn:prob.update.two.u}, \\
\sum_{*\in RP}f_{*,v,i-1} \cdot \frac{1-\rho_i'}{|RP|}\cdot \rho_i', & v \text{ recombined} \label{eqn:prob.update.two.v}, \\
\sum_{*,*\in RP \cdot RP} f_{*,*,i-1} \cdot\frac{1-\rho_i'}{|RP|} \cdot \frac{1-\rho_i'}{|RP|} ), & \text{both recombined}.
\end{numcases}
If it both recombined, sample the path, according $P(p_1 = u, p_2 = v) = f(u,v,i-1)$. If one of them recombined, sample the path according to the marginal probability of $P(p_1 = u) = f(u,i-1)$.

%\end{enumerate}

%{\bf Note:} In order to make greater variations between strains, we forbid two strains to copy from the same haplotype. Hence, $P_{i}(g_{p_1},g_{p_2}\big{|}D) = 0$ when $p_1 = p_2$.
%\begin{enumerate}
%\item At equation \eqref{eqn:prob.update.two}, $f_{u,v,i} = 0$, when $u = v$.

%\item Change equations \eqref{eqn:prob.update.two.u} and \eqref{eqn:prob.update.two.v} to
%\begin{numcases}\\
%\sum_{*\in R\setminus u}f_{u,*,i-1} \cdot\frac{1-\rho_i'}{|RP|}\cdot p.no.recomb  & u \text{ recombined} \label{eqn:prob.update.two.u2} \\
%\sum_{*\in R\setminus v}f_{*,v,i-1} \cdot \frac{1-\rho_i'}{|RP|} \cdot p.no.recomb  & v \text{ recombined} \label{eqn:prob.update.two.v2}
%\end{numcases}
%\end{enumerate}

\item
Ultermately, we consider add miss copies similar to the previous section, and sample the strain state given the path state with probabilities:
\begin{equation}
P(g_{s_1} = *, g_{s_2} = \# \big{|} D) =
\begin{cases}
P(g_{p_1} = *, g_{p_2} = \# \big{|} D) \cdot (1-\mu) \cdot (1-\mu), & g_{s_1} = g_{p_1} \text{ and } g_{s_2} = g_{p_2} ;\\
P(g_{p_1} = *, g_{p_2} = 1-\# \big{|} D) \cdot (1 - \mu) \cdot \mu, & g_{s_1} = g_{p_1} \text{ and } g_{s_2} \neq g_{p_2};\\
P(g_{p_1} = 1-*, g_{p_2} = \# \big{|} D) \cdot \mu \cdot (1 - \mu), & g_{s_1} \neq g_{p_1} \text{ and } g_{s_2} = g_{p_2};\\
P(g_{p_1} = 1-*, g_{p_2} = 1-\# \big{|} D) \cdot \mu \cdot \mu, & g_{s_1} \neq g_{p_1} \text{ and } g_{s_2} \neq g_{p_2}.
\end{cases}
\label{eqn:ps0}
\end{equation}
\end{enumerate}




\begin{itemize}
\item Randomly update a single strain or two strains simultaneously.\\
A crucial difference between our method and LS's model is that mixed samples can have more than two strains, with unknown proportions. We randomly choose the strains to update, then apply LS's algorithm to sample the path (see Fig.~\ref{fig:ls}) using Gibbs sampler with given proportion rather than 50/50 in the cases of diploid samples.
\item Updating the haplotypes from the paths, take into account of miss copying. Our model benefits from combining information from both the reference haplotypes as well as the data. For {\em de novo} mutations which are not found reference panel, our method will infer mutations based on read count from data.


\end{itemize}


\subsection{Model selection}
The final iteration of the MCMC sample is treated as a point estimate to infer the haplotypes and proportion. %\todo{DIC calculation}
Hence the deconvolution process is repeated with different random seeds; the best fit model is chosen according to the lowest deviance information criterion. The DIC is calculated from the samples generated by a Markov chain Monte Carlo simulation, and penalized by the average diviance.
    %DIC.WSAF.bar = -2 * sum(thetallk)
    %return (  mean(-2*tmpllk) + (mean(-2*tmpllk) - DIC.WSAF.bar) ) # D_bar + pD, where pD = D_bar - D_theta, and D_bar = mean(D_theta)



\section{Validation}
A set of {\it in vitro} mixtures of parasites were created by \citet{Wendler2015} to simulate mixed infection, which is an ideal validation data set in our use. In this data set, DNA was extracted from four laboratory parasite lines: 3D7, Dd2, HB3 and 7G8, and mixed with different ratios of mixed infection (see Table~\ref{tab:jason} in brackets), and submitted to the MalariaGEN pipeline \citep{MalariaGen2008} for Illumina sequencing.


\begin{table}[ht]\centering
\begin{tabular}[c]{@{}l|llll@{}}\hline
sample    & 3D7 & Dd2 & HB3 & 7G8 \\ \hline
{\tt	PG0389-C}	&	88.5	(90)	&	11.5	(10)	&		0	&		0	\tabularnewline
{\tt	PG0390-C}	&	79.8	(80)	&	20.2	(20)	&		0	&		0	\tabularnewline
{\tt	PG0391-C}	&	66.1	(67)	&	33.9	(33)	&		0	&		0	\tabularnewline
{\tt	PG0392-C}	&	31.2	(33)	&	68.8	(67	&		0	&		0	\tabularnewline
{\tt	PG0393-C}	&	18.4	(20)	&	81.6	(80)	&		0	&		0	\tabularnewline
{\tt	PG0394-C}	&	9.1	(10)	&	90.1	(90)	&		0	&		0	\tabularnewline
{\tt	PG0395-C}	&		0	&	33.6	(33.3)	&	35	(33.3)	&	31.3	(33.3)	\tabularnewline
{\tt	PG0396-C}	&		0	&	25.9	(25)	&	26.1	(25)	&	48	(50)	\tabularnewline
{\tt	PG0397-C}	&		0	&	14.7	(14.3)	&	15.3	(14.3)	&	69.9	(71.4)	\tabularnewline
{\tt	PG0398-C}	&		0	&		0	&	45.1+54.9	(100)	&		0	\tabularnewline
{\tt	PG0399-C}	&		0	&		0	&	56.7+40.9	(99)	&	2.4	(1)	\tabularnewline
{\tt	PG0400-C}	&		0	&		0	&	39.5+57.5	(95)	&	3	(5)	\tabularnewline
{\tt	PG0401-C}	&		0	&		0	&	33.3+56.7	(90)	&	10	(10)	\tabularnewline
{\tt	PG0402-C}	&		0	&		0	&	85.2	(85)	&	14.8	(15)	\tabularnewline
{\tt	PG0403-C}	&		0	&		0	&	80.1	(80)	&	19.3	(20)	\tabularnewline
{\tt	PG0404-C}	&		0	&		0	&	75.4	(75)	&	24.6	(25)	\tabularnewline
{\tt	PG0405-C}	&		0	&		0	&	70.6	(70)	&	29.4	(30)	\tabularnewline
{\tt	PG0406-C}	&		0	&		0	&	61	(60)	&	39	(40)	\tabularnewline
{\tt	PG0407-C}	&		0	&		0	&	50.5	(50)	&	49.5	(50)	\tabularnewline
{\tt	PG0408-C}	&		0	&		0	&	40.1	(40)	&	59.2	(60)	\tabularnewline
{\tt	PG0409-C}	&		0	&		0	&	30.1	(30)	&	69.1	(70)	\tabularnewline
{\tt	PG0410-C}	&		0	&		0	&	25.9	(25)	&	73,4	(75)	\tabularnewline
{\tt	PG0411-C}	&		0	&		0	&	21.4	(20)	&	78.5	(80)	\tabularnewline
{\tt	PG0412-C}	&		0	&		0	&	15.2	(15)	&	84.8	(85)	\tabularnewline
{\tt	PG0413-C}	&		0	&		0	&	3.8	(5)	&	96.2	(95)	\tabularnewline
{\tt	PG0414-C}	&		0	&		0	&	0	(1)	&	29.9+70.1	(99)	\tabularnewline
{\tt	PG0415-C}	&		0	&		0	&		0	&	30.0+70.0	(100)	\tabularnewline
\hline
\end{tabular}
\caption{Inferred percentages (true in brackets) of the mixed samples.}
\label{tab:jason}
\end{table}

We apply our program to 27 lab-mixed {\em in vitro} samples to validate our methods and program. As discribed in section~\ref{sec:updateP}, we start our method with the assumption of at most three strains present in the mixtures; and discard the strains less than 1\%. Our method successfully recovers the proportions with haplotypes of the input (see Table~\ref{tab:jason}). The differences between our proportion estimates and the truth is at most 2\%. Detailed validation steps are described in the supplement material.

Our model overfits the noisy lab-mixed sample with additional strains. Note that in Table~\ref{tab:jason}, we infer six of the HB3 and 7g8 mixtures as mixing of three, two of which haplotypes have subbtle difference with the same parasite line, but overall vastly different the last strain. The subbtle variation is caused by few hetrozigous sites with high coverage resulting high leverage in our model (see supplemental Figure~S4.3(a)). The origin of the noisy markers are possibily from sequncing or variant calling process, which are not recalabriated by our program.


We experimented with using different reference panels for deconvolution. In all cases we estimated the number and proportion of strains accurately, but the accuracy of the haplotype is dependent on having an approriate reference panel (see Figure~\ref{fig:differentRefPanel}).


\begin{figure}[h]
\centerline{
\includegraphics[width=0.45\textwidth]{{differentPanelForSample.PG0396-C}.png}
}
\caption{Haplotypes comparison of sample {\tt PG0396-C} chromosome 14 deconvolution without any reference strain (top) versus with using different reference panels (from the second to the bottom): I. 5 Asian and 5 African clonal strains; II. panel I with the addition of HB3; III. panel II with the addition of 7g8; IV. panel III with the addition of dd2. Black bars indicate alternative alleles; red bars mark wrongly inferred positions. The yellow, cyan and white background label the haplotype segements from strains 7g8, HB3 and dd2 respectively.}\label{fig:differentRefPanel}
\end{figure}

\section{Performance}
\subsection{Accuracy}

\begin{figure}[h]%figure1
\centerline{
\includegraphics[width=0.45\textwidth]{{switchVsMisCopyErrlog}.png}
}
\caption{}\label{fig:swithVsMisCopy}
\end{figure}

%\begin{table}
%\centering
%\begin{tabular}{c|c|c}\hline
%Panel & siwtch error & miss copy error \\\hline
%No panel &   67 &  1718\\
%I  & 38   &  1314\\
%II  &  24  &  773\\
%III &   12 &   450\\
%IV &   0 &   146\\\hline
%\end{tabular}
%\caption{}\label{tab:differentRefPanel}
%\end{table}



%\begin{figure}[ht]
%\begin{center}
%\missingfigure[figwidth=6cm]{Haplotype quality when the perfect panel was used!!!}
%\end{center}
%\end{figure}

\subsection{Run-time}
The complexity of our program is $\mathcal{O}(n^2m)$, where $n$ and $m$ are the number of reference strains and sites respectively. In practice, we divide Pf3k samples into several geographical region and perform deconvolution, with ten most diversed clonal strains as reference panel. The run time of deconvoluting a field sample range from 1 to 6 hours. For example, it takes $5\frac{1}{2}$ hours to process sample {\tt QG0182-C} with 372,884 sites.

%94m31.924s PD0500-C 98,211
%328m52.036s  QG0182-C 372,884

\begin{figure}[ht]
\centering
\includegraphics[width=.5\textwidth,height=0.4\textwidth]{runTime.png}
\end{figure}

\end{methods}


\section{Discussion}
The program DEploid and its analysis pipeline is originally developed for {\it P. falciparum} studies. With some specific minor parameter changes, DEploid can be used for deconvolute {\it P. vivax} sequence data \citep{Pearson2016}. The framework is suitable for deconvoluting mixed genomes with unknown proportions. It can thus be extended to a wider range of applications, such as deconvoluting cancer tumour cell genomes or Ebola virus genomes.



\section*{Acknowledgements}
We thank valuable insights and suggestions from Roberto Amato, John O'Brien, Richard Pearson, and Jason Wendler for providing the data of artificial samples. We thank Zam Iqbal for naming the program DEploid.

\section*{Funding}
This project is funded by the Wellcome Trust grant [100956/Z/13/Z].\\
~\\
\noindent{\em Conflict of Interest: none declared.}


\begin{thebibliography}{}

\bibitem[\protect\citeauthoryear{Arnot}{Arnot}{1998}]{Arnot1998}
Anita, D. (1998).
\newblock Unstable malaria in Sudan: the influence of the dry season: clone
  multiplicity of {\it Plasmodium falciparum} infections in individuals exposed to
  variable levels of disease transmission.
\newblock {\em Transactions of The Royal Society of Tropical Medicine and
  Hygiene\/}~{\em 92\/}(6), 580--585.

\bibitem[\protect\citeauthoryear{Browning and Browning}{Browning and
  Browning}{2007}]{Browning2007}
Browning, S.~R. and B.~L. Browning (2007).
\newblock Rapid and accurate haplotype phasing and missing-data inference for
  whole-genome association studies by use of localised haplotype clustering.
\newblock {\em The American Journal of Human Genetics\/}~{\em 81\/}(5), 1084--1097.

\bibitem[\protect\citeauthoryear{de~Roode, Culleton, Bell, and Read}{de~Roode
  et~al.}{2004}]{deRoode2004}
de~Roode, J., R.~Culleton, A.~Bell, and A.~Read (2004).
\newblock Competitive release of drug resistance following drug treatment of
  mixed {\it Plasmodium Chabaudi} infections.
\newblock {\em Malaria Journal\/}~{\em 3\/}(33), 1--6.

\bibitem[\protect\citeauthoryear{de~Roode, Pansini, Cheesman, Helinski,
  Huijben, Wargo, Bell, Chan, Walliker, and Read}{de~Roode
  et~al.}{2005}]{deRoode2005}
de~Roode, J.~C., R.~Pansini, S.~J. Cheesman, M.~E.~H. Helinski, S.~Huijben,
  A.~R. Wargo, A.~S. Bell, B.~H.~K. Chan, D.~Walliker, and A.~F. Read (2005).
\newblock Virulence and competitive ability in genetically diverse malaria
  infections.
\newblock {\em Proceedings of the National Academy of Sciences of the United
  States of America\/}~{\em 102\/}(21), 7624--7628.

\bibitem[\protect\citeauthoryear{Galinsky}{Galinsky et~al.}{2015}]{Galinsky2015}
Galinsky, K., Valim, C., Salmier, A., de Thoisy, B., Legrand, E., Faust, A., Baniecki, M. L., Ndiaye, D., Daniels, R. F., Hartl, D. L., Sabeti, P. C., Wirth, D. F., Volkman, S. K., Neafsey, Daniel E.(2015).
\newblock COIL: a methodology for evaluating malarial complexity of infection using likelihood from single nucleotide polymorphism data.
\newblock {\em Malaria Journal\/}~{\em14\/}(4), 1--9.

\bibitem[\protect\citeauthoryear{Hastings and D\'Alessandro}{Hastings and
  D\'Alessandro}{2000}]{Hastings2000}
Hastings, I. and U.~D'Alessandro (2000).
\newblock Modelling a predictable disaster: the rise and spread of
  drug-resistant malaria.
\newblock {\em Parasitology Today\/}~{\em 16\/}(8), 340--347.

\bibitem[\protect\citeauthoryear{Howie, Donnelly, and Marchini}{Howie
  et~al.}{2009}]{Howie2009}
Howie, B.~N., P.~Donnelly, and J.~Marchini (2009).
\newblock A flexible and accurate genotype imputation method for the next
  generation of genome-wide association studies.
\newblock {\em PLoS Genet\/}~{\em 5\/}(6), 1--15.

\bibitem[\protect\citeauthoryear{Li and Stephens}{Li and
  Stephens}{2003}]{Li2003}
Li, N. and M.~Stephens (2003).
\newblock {Modeling linkage disequilibrium and identifying recombination
  hotspots using single-nucleotide polymorphism data}.
\newblock {\em Genetics\/}~{\em 165\/}(4), 2213--2233.

\bibitem[\protect\citeauthoryear{MalariaGEN}{MalariaGEN}{2008}]{MalariaGen2008}
MalariaGEN (2008).
\newblock A global network for investigating the genomic epidemiology of
  malaria.
\newblock {\em Nature\/}~{\em 456\/}(7223), 732 -- 737.

\bibitem[\protect\citeauthoryear{Pearson, Amato, Auburn, Miotto,
  Almagro-Garcia, Amaratunga, Suon, Mao, Noviyanti, Trimarsanto, Marfurt,
  Anstey, William, Boni, Dolecek, Tran, White, Michon, Siba, Tavul, Harrison,
  Barry, Mueller, Ferreira, Karunaweera, Randrianarivelojosia, Gao, Hubbart,
  Hart, Jeffery, Drury, Mead, Kekre, Campino, Manske, Cornelius, MacInnis,
  Rockett, Miles, Rayner, Fairhurst, Nosten, Price, and Kwiatkowski}{Pearson
  et~al.}{2016}]{Pearson2016}
Pearson, R.~D., R.~Amato, S.~Auburn, O.~Miotto, J.~Almagro-Garcia,
  C.~Amaratunga, S.~Suon, S.~Mao, R.~Noviyanti, H.~Trimarsanto, J.~Marfurt,
  N.~M. Anstey, T.~William, M.~F. Boni, C.~Dolecek, H.~T. Tran, N.~J. White,
  P.~Michon, P.~Siba, L.~Tavul, G.~Harrison, A.~Barry, I.~Mueller, M.~U.
  Ferreira, N.~Karunaweera, M.~Randrianarivelojosia, Q.~Gao, C.~Hubbart,
  L.~Hart, B.~Jeffery, E.~Drury, D.~Mead, M.~Kekre, S.~Campino, M.~Manske,
  V.~J. Cornelius, B.~MacInnis, K.~A. Rockett, A.~Miles, J.~C. Rayner, R.~M.
  Fairhurst, F.~Nosten, R.~N. Price, and D.~P. Kwiatkowski (2016, June).
\newblock {Genomic analysis of local variation and recent evolution in
  Plasmodium vivax}.
\newblock {\em Nat Genet\/}~{\em 48}, 959--964.

\bibitem[\protect\citeauthoryear{Pf3k}{Pf3k}{2016}]{Pf3k2016}
The Pf3k Project: pilot data release 5 (2016).
%\newblock {\em The Pf3k Project: pilot data release 5}.
\newblock {www.malariagen.net/data/pf3k-5} [accessed 1 June 2016]

\bibitem[\protect\citeauthoryear{O'Brien}{O'Brien et~al.}{2015}]{Jack2016}
O'Brien D,J., Iqbal Z, Wendler J, Amenga-Etego L (2016). \newblock Inferring Strain Mixture within Clinical {\em Plasmodium falciparum} Isolates from Genomic Sequence Data. \newblock {\em PLoS Comput Biol\/}~{\em 12\/}(6): e1004824. doi: 10.1371/journal.pcbi.1004824


\bibitem[\protect\citeauthoryear{Wendler}{Wendler}{2015}]{Wendler2015}
Wendler, J. (2015).
\newblock {\em Accessing complex genomic variation in} {P}lasmodium falciparum {\em natural infection}.
\newblock Ph.\ D. thesis, University of Oxford.

%\bibitem{pfdeconv}
%Zhu, S., PfDeconv (2016).
%\newblock {GitHub repository}, \newblock {https://github.com/mcveanlab/PfDeconv}  [accessed 1 June 2016]

\end{thebibliography}
\end{document}
