\documentclass{bioinfo}
%\usepackage[colorinlistoftodos]{todonotes}
\usepackage[disable]{todonotes}


\newcounter{todocounter}
\newcommand{\todonum}[2][]
{\stepcounter{todocounter}\todo[#1]{\thetodocounter: #2}}

\newcommand{\done}[2][]
{\todo[color=green!40, #1]{#2}}

\newcommand{\donenum}[2][]
{\stepcounter{todocounter}\done[#1]{\thetodocounter: #2}}



\usepackage{xcolor}
\usepackage{hyperref}
\hypersetup{
    colorlinks,
    %linkcolor={red!50!black},
    linkcolor={black},
    citecolor={blue!50!black},
    urlcolor={blue!80!black}
}


\usepackage{colortbl}
\usepackage{cases}
\graphicspath{{./figures/}}
\definecolor{RubineRed}{RGB}{240, 0, 240}       % RubineRed  Approximate PANTONE RUBINE-RED


\newcommand*{\Scale}[2][4]{\scalebox{#1}{$#2$}}%



\copyrightyear{2016} \pubyear{2016}
\access{Advance Access Publication Date: Day Month Year}
\appnotes{Original Paper}


\begin{document}

\listoftodos
\clearpage
\setcounter{page}{1}

\firstpage{1}

\subtitle{Genome analysis}

\title[Deconvolution of multiple infections]{Deconvolution of multiple infections in {\it Plasmodium falciparum} from high throughput sequencing data}
\author[Zhu \textit{et~al}.]{Sha Joe Zhu\,$^{\text{\sfb 1,\textcolor{red}{2}},*}$, Jacob Almagro-Garcia\,$^{\text{\sfb 1,\textcolor{red}{2},3,4}}$ and Gil McVean\,$^{\text{\sfb 1,\textcolor{red}{2}},*}$}
\address{
$^{\text{\sf 1}}$ Wellcome Trust Centre for Human Genetics, University of Oxford, Oxford, UK \\
\textcolor{red}{
$^{\text{\sf 2}}$ Big Data Institute, Li Ka Shing Centre for Health Information and Discovery, University of Oxford, Oxford, UK\\
$^{\text{\sf 3}}$ Medical Research Council (MRC) Centre for Genomics and Global Health, University of Oxford, Oxford, UK \\
$^{\text{\sf 4}}$ Wellcome Trust Sanger Institute, Hinxton, UK \\
}
\donenum[inline]{updated affiliation}
}

\corresp{$^\ast$To whom correspondence should be addressed.}

\history{Received on \textcolor{red}{XXXXX}; revised on \textcolor{red}{XXXXX}; accepted on \textcolor{red}{XXXXX}}

\editor{Associate Editor: \textcolor{red}{XXXXXXX}}

\abstract{\textbf{Motivation:}
The presence of multiple infecting strains of the malarial parasite {\it Plasmodium falciparum} affects  key phenotypic  traits, including drug resistance and risk of severe disease. Advances in protocols and sequencing technology have made it possible  to obtain  high-coverage genome-wide  sequencing data from blood samples and blood spots taken in the field. However, analysing and interpreting such data is challenging  because of the high rate of multiple infections present.\\
\textbf{Results:} We have developed a statistical method and implementation for deconvolving multiple genome sequences present in an individual with mixed infections.  The software package {\it DEploid} uses haplotype structure within a reference panel of clonal isolates as a prior for haplotypes present in a given sample. It estimates the number of strains, their relative proportions and the haplotypes presented in a sample, allowing researchers to study multiple infection in malaria with an unprecedented level of detail.\\
\textbf{Availability and implementation:} The open source implementation {\it DEploid} is freely available at \href{https://github.com/mcveanlab/DEploid}{https://github.com/mcveanlab/DEploid} under the conditions of the GPLv3 license. An R version is available at \href{https://github.com/mcveanlab/DEploid-r}{https://github.com/mcveanlab/DEploid-r}.\\
\textbf{Contact:} \href{joe.zhu@well.ox.ac.uk}{joe.zhu@well.ox.ac.uk} or \href{mcvean@well.ox.ac.uk}{mcvean@well.ox.ac.uk}\\
\textbf{Supplementary information:} Supplementary data are available at \textit{Bioinformatics} online.}

\maketitle

\section{Introduction}
Malaria remains one of the top global health problems. The majority of malaria related deaths are caused by the {\it Plasmodium falciparum} parasite \citep{WHO2016}, transmitted by mosquitoes of the genus {\it Anopheles}. Patients are often infected with more than one distinct parasite strain (termed mixed infection, multiple infection, or complexity of infection), due to bites from multiple mosquitoes, mosquitoes carrying multiple genetic types or a combination of both. Mixed infections  can lead to competition among co-existing strains and may influence disease development \citep{deRoode2005}, transmission rates \citep{Arnot1998} and the spread of drug resistance \citep{deRoode2004}. In addition, within-host evolution can lead to the presence of more than one genetically and phenotypically distinct strains \citep{Bell2006}.

The presence of multiple strains of {\it P. falciparum} makes fine scale analysis of genetic variation challenging, since genetic differences between strains of this haploid organism will appear as heterozygous loci. Such mixed calls confound methods that exploit haplotype data to detect, among other phenomena, the occurrence of natural selection or recent demographic events \citep{Harris2013, Lawson2012, Mathieson2014, Sabeti2002}. In light of these difficulties, researchers usually focus on clonal infections or resort to heuristic methods for resolving heterozygous genotypes. The former approach discards valuable information regarding genetic diversity and relatedness, whereas the latter tends to create chimeric haplotypes that are not suitable for analysis, unless mixed calls are very sparse.

In comparison to the problem of phasing haplotypes within diploid organisms, deconvolving the strains of a multiple infection differs because of uncertainty in the number of strains present and their relative proportions.  Consequently, existing tools for phasing diploid organisms, such as BEAGLE \citep{Browning2007}, IMPUTE2 \citep{Howie2009} and SHAPEIT \citep{Delaneau2012, Oconnell2014}, are not appropriate. \citet{Galinsky2015}, \citet{Jack2016} \donenum{REV1.20: O'Brien (2016)} \textcolor{red}{and} \citet{Chang2017} have attempted to address the multiple infection problem by inferring the number and proportions of strains from allele frequencies within samples.  However, since they do not infer haplotypes, these approaches have limited applicability.

As part of the Pf3k project \citep{Pf3k2016}, an effort to map the genetic diversity of {\it P. falciparum} at global scale, we have developed algorithms and a software package implementation \texttt{DEploid}, for deconvolving multiple infections. The program estimates the number of different genetic types present in the isolate, the proportion or abundance of each strain and their sequences (i.e. haplotypes). To our knowledge, \texttt{DEploid} is the first package able to deconvolute strain haplotypes and provides a unique opportunity for researchers to study the epidemiology of {\it P. falciparum}.


\section{Methods}

\subsection{Notations}

We first introduce our notation (see Table~\ref{tab:notation}). Our data, $D$, are the allele read counts of sample $j$ at a given site $i$, denoted as $r_{j,i}$ and $a_{j,i}$ for reference (REF) and alternative (ALT) alleles respectively.  These are assigned values of $0$ and $1$ resepctively. Here we consider only biallelic loci, though future extension to include multi-allelic sites is simple.  The empirical allele frequencies within a sample (WSAF) $p_{j,i}$ and at population level (PLAF) $f_i$ are calculated by $ \frac{a_{j,i}}{a_{j,i} + r_{j,i}}$ and $ \frac{\sum_j a_{j,i}}{\sum_j a_{j,i} + \sum_j r_{j,i}}$ respectively. Since all data in this section refers to the same sample, we drop the subscript $j$ from now on.

\begin{table}[htb]\centering
\begin{tabular}{c|c}\hline
$i$              & Marker index\\
$j$              & Sample index \\
$r$              & Read count for reference allele \\
$a$              & Read count for alternative allele \\
$f$              & Population level allele frequency (PLAF) \\
$n$              & Number of strains within sample \\
$l$              & Sequence length \\
$\mathbf{w}$      & Proportions of strains \\
$\mathbf{x}$	& Log titre of strains \\
$\mathbf{h}_{i}$ & Allelic states of $n$ parasite strains at site $i$ \\
$h_{k,i}$   & Allelic state of parasite strain $k$ at site $i$\\
$p$              & Observed within sample allele frequency (WSAF) \\
$q$              & Unadjusted expected WSAF  \\
$\pi$            & Adjusted expected WSAF \\
$\Xi$            & Reference panel\\
$\xi_{k,i}$     & Allelic state of reference panel strain $k$ at site $i$\\
$G$              & Scaling factor used for genetic map\\
$e$              & Probability of read error\\ \hline
\end{tabular}
\caption{Table summarising the notation used in this article.}\label{tab:notation}
\end{table}


\subsection{Model}

We describe the mixed infection problem by considering the number of strains, $n$, the relative abundance of each strain, $\mathbf{w}$, and their allelic states, $\mathbf{h}$. Similar to \citet{Jack2016}\donenum{REV1.20: O'Brien (2016)}, we use a Bayesian approach and define the posterior probabilities of $n$, $\mathbf{w}$ and $\mathbf{h}$ given a reference panel, $\Xi$, and the read error rate, $e$, as:

\begin{equation}
P(n, \mathbf{w}, \mathbf{h}, | \Xi, e, D) \propto L(n, \mathbf{w}, \mathbf{h}, | \Xi, e, D) \times P(n, \mathbf{w}, \mathbf{h}). \label{eqn:post}
\end{equation}

\noindent We assume a prior in which the haplotypes of the $n$ strains are independent of each other and dependent only on the reference panel.  Therefore, the joint prior can be written as:

\begin{equation}
P(n, \mathbf{w}, \mathbf{h}) = P(n) \times P(\mathbf{w} | n) \times \prod_{k=1}^{n} P(h_k | \Xi).
\end{equation}

\noindent The following sections describe details of the model and the approach to inference.


\subsubsection{Likelihood function}

Let $\mathbf w = [w_1,\dots, w_n]$ and $\mathbf{h}_i = [h_{1,i},\dots,h_{n,i}]$ denote the proportions and alleic states of the $n$ parasite strains at site $i$. We use \citet{Jack2016}\donenum{REV1.20: O'Brien (2016)}'s expression for the expected WSAF at site $i$, $q_{i}$, as:

\begin{equation}
q_i= (\mathbf{w}\cdot\mathbf{h}_{i})  =  \sum_{k=1}^{n} w_k \cdot h_{k,i} .\label{eqn:qij_full_sum}
\end{equation}

\noindent The data, which can be summarised by the reference and alternative allele read counts at each site, is modelled through a beta-binomial distribution given the expected WSAF.  We model the data at distinct segregating sites as independent.  Thus the likelihood function  in Eqn.~\eqref{eqn:post} is only dependent on the haplotypes present and their frequencies through their contribution to $q_{i}$.

%; i.e. $L(n, \mathbf{w}, \mathbf{h}, | \Xi, e, D) = \prod_{i=1}^l L(q_i | \Xi, e, D)$.

To incorporate sequencing error, we modify the expected WSAF such that the allele frequency of `REF' read as `ALT' is $(1 - q_i)e$, and the allele frequency of `ALT' read as `REF' is $q_ie$. Thus, the adjusted expected WSAF becomes:

\begin{equation}
\pi_i = q_i + (1 - q_i)e - q_ie = q_i + (1 - 2q_i)e.\label{eqn:adj_q}
\end{equation}

\noindent We model over-dispersion in read counts relative to the Binomial using a Beta-binomial distribution. Specifically, the read counts of `ALT' are identically and independently distributed (i.i.d.) Bernoulli random variables with probability of success $v_i$; i.e. $a_i \sim Binom(a_i + r_i, v_i)$, and $v_i \sim Beta(\alpha, \beta)$, where $E(v_i) = \alpha/(\alpha+\beta) = \pi_{i}$. This is achieved by setting $\alpha = c\cdot \pi_{i} $ and $\beta = c\cdot (1-\pi_{i})$, such that the variance of the WSAF is \textcolor{red}{inversely} \donenum{REV3.6: sp inversley} proportion to $c$. Combined, we have:

\begin{equation}
L(q_{i}| e, D) \propto \frac{\Gamma(a_i + c\cdot \pi_{i}) \Gamma(r_i + c\cdot (1-\pi_{i}))}{\Gamma(c\cdot \pi_{i})\Gamma(c\cdot (1-\pi_{i}))}. \label{eqn:llk}
\end{equation}


\subsubsection{Prior distributions}\label{sec:prior}

Rather than model the number of strains, $n$, directly, we take the approach of fixing $n$ to be at the upper end of what can realistically be inferred (typically $5$), using a skewed prior for proportions (such that typically only $1-2$ strains might be at appreciable frequency) and then discarding strains inferred to have a proportion less than some critical amount (e.g. 1 percent).

To achieve this, we model the proportions of the $n$ strains through a log titre, $x_k$, drawn from a $N(\eta, \sigma^2)$ prior.  The proportion of strain $k$, $w_k$, is given by

\begin{equation}
w_k = \frac{\exp(x_k)}{\sum_{j=1}^n \exp(x_j)},
\end{equation}

\noindent and the prior density is given by the distribution function for the value of $\mathbf{x}$.

Haplotypes, $\mathbf{h}$, are modelled as being generated independently from the reference panel by the \citet{Li2003} process, though with a rate of mis-copying that is independent of the panel size. That is, under the prior, a path through the reference panel is sampled as a Markov process where recombination enables switching between members of the reference panel and mis-copying allows the allelic state of the haplotype within the sample to differ from the allelic state of the reference panel haplotype being copied at the site.  The transition probability of switching from copying reference haplotype $a$ to reference haplotype $b$ is $(1-\exp(-G \psi_i))/|\Xi|$, where $\psi_i$ is the genetic distance (in Morgans) between sites $i$ and $i+1$, $G$, is a scaling factor (described below in more detail) and $|\Xi|$ is the size of the reference panel.  Note that unlike the original model, the recombination or switching rate is not dependent on sample size.


For miscopying, let $\xi_k$ denote the state of the sequence in the reference panel $\Xi$ that $h_k$ is copying from at given site and $\mu$ denote the probability of miss-copying:

$$
\begin{cases}
P(\xi_k = h_k) &= 1-\mu, \\
P(\xi_k \neq h_k) &= \mu.
\end{cases}
$$

\noindent As above, this is a simple reparamterisation of the original model, but where the miscopying rate is independent of the sample size. The emission probabilities are given by the convolution of the reference panel paths and the miscopying process, strain proportions and the read error rate.




\subsection{Inference}
\textcolor{red}{
To perform inference about the haplotypes present and their proportions we use Markov chain Monte Carlo (MCMC). We use signatures of allele frequency imbalance to learn the relative abundance using a Metropolis-Hastings algorithm, which samples proportions ($\mathbf w$) given $\mathbf h$. While updating the proportion, we use the haplotype structure provided by the reference panel to shape the inferred haplotype, which uses a Gibbs sampler to update $\mathbf h$ for a given $\mathbf w$, with two types of update: a single haplotype and a pair of haplotypes. These MCMC updates are iterated in a random order. Details can be found in the supplementary materials.}
\donenum{REV1.17: Pair updates always after single update?}



\subsection{Implementation details}

\begin{itemize}
\item {\bf Number of strains}. As described above, we aim to infer more strains than are actually present, starting the MCMC chain with a fixed $n$, which has a default of $5$. At the point of reporting, we discard strains with a proportion less than a fixed threshold, typically $0.01$.

\item {\bf Parameters}. \textcolor{red}{The parameter $c$ (Equation~\eqref{eqn:llk}) reflects how much data is available. The mean coverage of the validation data set ranges from 106.20 to 147.04, with a mean of 124.487.} \donenum{REV3.5: what is c?} In practice, we set the parameters $c=100$; $\eta = 0$, \textcolor{red}{$\sigma^2 = 5$ which are adjusted accordingly when working with extremely unbalanced samples (Sections \ref{sec:prior} and supplementary material)}.  We set the read error rate as 0.01 and the rate of mis-copying as 0.01.

\item {\bf Recombination rate and scaling}. We assume a uniform recombination map, where the genetic distance between loci $i$ and $i+1$ is computed by $\psi_i = D_i / d_m$ where $D_i$ denotes the physical distance between loci $i$ and $i+1$ in nucleotides and $d_m$ denotes the average recombination rate in Morgans bp$^{-1}$. We use the recombination rate for {\it P. falciparum} of 15,000 base pairs per centiMorgan as reported by \citet{Miles2016}. The recombination rate is scaled by a factor $G$, which reflects the effective population size, rate of inbreeding and size and relatedness of the reference panel. \textcolor{red}{In practice, we deconvolve over 1 million markers of field samples, we use a value of $G=20$ to ensure small values for recombination probabilities between two markers, with a mean of 0.015. A large value of $G$ relaxes the reference panel constrain: It is effectively an LD free model when $G$ is infinity.} \donenum{REV3.2 explain value of G=20.} The scaled genetic distance $G\psi$ is used to compute the transition probability of switching from copying reference haplotype a to reference haplotype b (see Supplementary Materials for details).

\item {\bf Update without linkage disequilibrium}. For initialising the chain, or if the markers present are very widely spaced, linkage disequilibrium can be ignored, which is equivalent to setting the genetic distance between adjacent loci to be infinitely high.  Under these circumstances, the haplotype updates become much simpler and depend only on the population-level allele frequency (PLAF), for example as estimated from the reference panel or provided independently.

\item {\bf Reporting} We aim to provide users with a single point estimate of the haplotypes and their proportions, although the full chain is also available for analysis.  To achieve this we report values at the last iteration - i.e. we report a single sample from the posterior.  However, to measure robustness, we also typically repeat deconvolution with multiple random starting points\textcolor{red}{. We use the majority rule on inferred number of strains and keep the consensus chains; we then select the chain with the lowest average deviance (after removing the burn-in) to report. The deviance measures the difference in log likelihood between the fitted and saturated models, the latter being inferred by setting the WSAF to that observed.} These parameters can be modified by users to achieve a preferred balance between computational speed and confidence.  By default, we set the MCMC sampling rate as 5, with the first 50\% of samples removed as burn in and 800 samples used for estimation.

\item {\bf Reference panel construction}. To infer clonal samples for the reference panel we use the Pf3k \citep{Pf3k2016} project data, running the algorithm without LD on all samples and identifying those with a dominant haplotype (proportion > 0.99) as clonal.  These clonal samples are grouped by region of sampling to form location-specific reference panels.  In addition, we have included a number of reference strains, described in more detail below.

\end{itemize}



\section{Validation and Performance}

As validation we used a set of {\it in vitro} mixtures created by \citet{Wendler2015} to simulate mixed infections. DNA was extracted from four laboratory parasite lines: 3D7, Dd2, HB3 and 7G8, experimentally mixed in different proportions (see \textcolor{red}{Figure~\ref{fig:jason}}), and submitted to the MalariaGEN pipeline \citep{MalariaGen2008} for Illumina sequencing and genotyping \citep{Manske2012}.

This data set only contains two unmixed samples, which is insufficient for constructing a reference panel. Moreover, the {\em P. falciparum} genetic crosses project \citep{Miles2016} found that due to sequencing error, mapping error and variation among variant calling methods, genotype calls vary at the same locus for the same strain of {\it P. falciparum}. To create a baseline reference haplotype for each strain we therefore considered multiple samples that contains the same parasite strains.

\begin{figure}[htb]
\centering
\includegraphics[width=0.48\textwidth]{deconv/effectiveK/eff_k_both.pdf}
\caption{\textcolor{red}{Experimental validation and effective number of strains inferred from the \texttt{DEploid} method. We use the reference panel V to deconvolve the same lab-mixed samples, by starting to assume 3 and 5 strains within a sample. Each inference is then repeated without using a reference panel for comparison. Crosses in black indicate the true effective number of strains. Red crosses indicate inference result that we conclude from 30 replicates when using panel and assuming that the possible number of strains is 5. The coloured dots show the inferred effective number of strains, where dots in faded colours show inference from multiple runs. Overall, we show consistence inference when assuming different number of strains, with or without a reference panel; except one case of mixture of three with equal proportions. Without a reference panel, the method completely misinterpret the data as a mixing of two strains of proportions 1/3 and 2/3, which stresses the importance of using a reference panel during deconvolution.}}
\label{fig:jason}
\donenum[inline]{REV2.1: mcmc rerun}
\donenum[inline]{REV1.8: How many reference strains were used}
\end{figure}


\paragraph{Inferring haplotypes for Dd2 strain.}
Since 3D7 is the reference strain, we assume that strain Dd2 is the only source of `ALT' reads in samples {\textmd PG0389-C} to {\textmd PG0394-C}. Assuming markers are independent from each other, let $y$ be the read count for `ALT' allele and $x$ be the total read count weighted by the Dd2 mixing proportion (see \textcolor{red}{Figure~\ref{fig:jason}} in brackets), we use a regression model ($y = \beta_0 + \beta_{1} x$) to infer the Dd2 genotype: 1 if $\beta_{1}$ is significant with $p$-values below $0.001$; 0 otherwise.


\paragraph{Inferring haplotypes for HB3 and 7G8.}
Similarly, for samples {\textmd PG0398-C} to {\textmd PG0415-C}, we let variables $x_1$, $x_2$ be the coverages weighted by the mixing proportions of HB3 and 7G8 respectively; we use a regression model ($y = \beta_0 + \beta_{1} x_1 + \beta_{2} x_2$) to infer the genotypes of HB3 and 7G8: HB3 is 1 if $\beta_{2}$ is significant with $p$-values below $0.001$; 0 otherwise; similarly for 7G8.

\textcolor{red}{To investigate how the accuracy of inference is affected by the quality of the reference panel (in terms of having haplotypes close to those present in the samples) we experimented with deconvolving the 27 lab-mixed samples with the following reference panels:
}
\begin{itemize}

\item \textcolor{red}{panel I: five Asian and five African clonal strains from the Pf3k \citep{Pf3k2016} resource: {\textmd PD0498-C}, {\textmd PD0500-C}, {\textmd PD0660-C}, {\textmd PH0047-Cx}, {\textmd PH0064-C}, {\textmd PT0002-CW}, {\textmd PT0007-CW}, {\textmd PT0008-CW}, {\textmd PT0014-CW}, {\textmd PT0018-CW};}

\item \textcolor{red}{panel II: panel I with the addition of HB3;}

\item \textcolor{red}{panel III: panel II with the addition of 7G8;}

\item \textcolor{red}{panel IV: panel III with the addition of Dd2;}

\item \textcolor{red}{panel V: 3D7, HB3, 7G8 and Dd2 strains (the perfect reference panel for the lab mixtures);}

\item \textcolor{red}{panel VI: panel I with the addition of six (three each) clonal strains from Asia and Africa: {\textmd PH0193-C}, {\textmd PH0283-C}, {\textmd PH0305}, {\textmd PT0060-C}, {\textmd PT0146-C} and {\textmd PT0158-C} (a typical reference panel for field samples of unknown geographical origin).}

\end{itemize}


\subsection{Accuracy}

\noindent \textcolor{red}{Our validation experiments use variant calls of these 27 lab-mixed {\it in vitro} samples, which are produced by the Pf3k pipeline \citep{Pf3k2016} based on GATK best practices \citep{McKenna2010} on 2512 field isolates and 128 lab samples. The filter threshold is set at a level for which false positive genotype calls (calling a variant that doesn't exist) and false negative calls (not calling a true variant) are equal. From the 18,570 high-quality biallelic SNPs, we observe a small number of heterozygous sites with high coverage, which can potentially mislead our model to over-fit the data with additional strains. After the filtering step (see supplementary materials for details), we deconvolve the remaining 17,530 sites for all experiments in the rest of this section, unless specified otherwise.}
\donenum{REV1.6:  How many SNPs?}
\donenum{REV1.16: filtering}
\donenum{REV3.8: filter erroneous markers}


\begin{figure*}[hbt]
\centerline{
\includegraphics[width=0.85\textwidth]{Fig2.pdf}
}
\caption{Comparison of true and inferred haplotypes for Chromosome 14 in sample {\textmd PG0396-C} without linkage disequilibrium (top) and using Reference Panels I to IV (from the second to the bottom). Reference Panel V gives results equivalent Panel IV and Panel VI gives results similar to Panel I.  Red bars mark wrongly inferred positions. The yellow, cyan and white background label the haplotype segments from strains 7G8, HB3 and Dd2 respectively. The switch errors are obtained by counting the changes of a strain segment mapped to reference strains; the genotype errors are the discordance between the strain and the mapped reference segments. \textcolor{red}{From the reference panel I to IV, as more relevant haplotype information is provided when deconvolving the haplotypes, it dramatically reduces inference errors in both switching and copying.}}
\label{fig:differentRefPanel}
\donenum[inline]{REV1.9: Figure 2: It would be useful to include how many SNPs were included for analysis on chromosome 14 in the figure legend.}
\donenum[inline]{REV3.3.1: black bars was confusing}
\donenum[inline]{REV3.3.2: I was not clear what the general take home message was meant to be}
\end{figure*}

\subsubsection{Proportions and number of strains}
\done{REV1.7: Why assume 3 strains but not 5?}
\noindent \textcolor{red}{Our method assumes a fixed number of strains present in the mixtures, and discards strains with an inferred proportion less than 1\%. In order to compare how reliable and robust when input more strains than we actually need, we introduce the effective number of strains, calculated by $1/\sum w_{i}^{2}$. The deconvolution experiments assume at most five (as default) and three strains (for comparison) within a sample. We find consistent inference of effective number of strains regardless the assumption of number of strains or with/without a reference panel (see Figure~\ref{fig:jason}). The deviance between the expected and inferred proportions per sample is bounded by the deviation between expected and observed effective number of strains inverse (derived in the supplementary material), with an average of 0.023. We explore the quality of proportion estimate from different reference panels of deconvolving a mixture of Dd2/7G8/HB3 three strains. In all cases we estimated the number and proportion of strains accurately, for example Figure~\ref{fig:differentRefPanel} shows the proportions of strains Dd2/7G8/HB3 as being accurately inferred as approximately $\frac{1}{4}$, $\frac{1}{2}$, and $\frac{1}{4}$. We find that deconvolution struggles with even-proportion mixtures without a reference panel, which provides necessary constrains, and enables our method to recover the right values.}


\begin{figure*}[htb]
\centerline{
\includegraphics[width=0.9\textwidth]{Fig3.pdf}
}
\caption{Comparison of \texttt{DEploid} and existing tools (\texttt{COIL}, \texttt{pfmix}, \texttt{BEAGLE}, and \texttt{SHAPEIT}). (a) Estimates for the number of strains present in each mixed infection (artificially mixed in the lab) as given by \texttt{COIL} and \texttt{DEploid}. (b) \textcolor{red}{Comparison of the inferred effective number of strains of each strain as given by \texttt{pfmix} and \texttt{DEploid}.} (c) Relationship between strain proportions and haplotype inference accuracy in the experimental validation for \texttt{DEploid} and \texttt{BEAGLE}/\texttt{SHAPEIT} (only mixtures of two strains). We use reference panel V to deconvolute all 27 samples. Each point represents a deconvoluted haplotype with 18,570 sites. Point shape refers to strain and colour indicates the method applied. \textcolor{red}{We use LOESS smoothing to show the trend of error vs. strain proportion.} Top panel shows switch error rate whereas the bottom panel indicates genotyping error rate. \textcolor{red}{Overall we find that \texttt{DEploid} inference results are comparable with existing methods on the number of strains and proportions; and \texttt{DEploid} can provide better results in haplotype inference, which is a significant advance in existing methods.}}
\label{fig:switchVsMisCopy}
\donenum[inline]{REV1.1: figure is far too small}
\donenum[inline]{REV3.4.1: "c is a noisy plot. It would be much clearer if shown with a smoothing."}
\donenum[inline]{REV3.4.2: It would inform the reader to say what the take home message of all plots should be in the legend.}
\donenum[inline]{REV1.14: cannot read the figure legends and axes.}
\end{figure*}


\subsubsection{Haplotypes}

Our accuracy assessment for inferred haplotypes takes into account both switch errors and genotype discordance, which reflects recombination and miscopying events. To understand how the inferred haplotypes relates to those present we split haplotypes into sets of 50 consecutive variants and assigned them to the reference strains through maximal identity.  Switches occur when adjacent segments of inferred haplotypes are closest to different reference strains.  Genotyping errors occur when a subset of sites within the segment differ from the closest reference strain.  Example deconvolutions are shown in Figure ~\ref{fig:differentRefPanel} and an overview of all experiments is shown in Figure ~\ref{fig:switchVsMisCopy}.  From our assessment of haplotype inference, we conclude:

\begin{itemize}

\item The inference of relative proportions does not seem to be affected by the use of linkage disequilibrium information from the reference panel or its closeness to the samples being analysed (Figure~\ref{fig:differentRefPanel}).

\item The accuracy of haplotype inference is, however, dependent on having an appropriate reference panel in terms of relatedness to the samples being analysed (Figure~\ref{fig:differentRefPanel}).

\item The strain proportion affects haplotype inference (see Figure~\ref{fig:switchVsMisCopy}). Our method infers strains with proportions over approximately 20\% with high accuracy, but struggles with minor strains due to insufficient data, in particular at sites when the minor strain carries the alternative allele and the dominant strain carries the reference allele (see Figure~\ref{fig:switchVsMisCopy}).

\end{itemize}


\subsection{Comparison to existing methods}
A mixed infection can be completely described by the number of co-existing strains, their relative proportions, and their associated haplotypes. Existing methods for characterizing mixed infections are limited to providing a summary statistic of relative inbreeding (F$_{\textrm{ws}}$, \cite{Manske2012}), inferring the number of strains (\texttt{COIL}), or simultaneously inferring the number of strains and their proportions (\texttt{pfmix}, \cite{Jack2016} \donenum{REV1.20: O'Brien (2016)}). DEploid is the only method that can also estimate haplotypes although it can be argued that conventional tools for phasing diploid organisms (\texttt{BEAGLE}, \texttt{SHAPEIT}) could be used to deconvolute mixtures of two strains.

\textcolor{red}{In this section, we use the same dataset (27 samples) to compare \texttt{DEploid} with all the inferential methods mentioned above (see Supplementary Material for details). Our method correctly infers the number of strains in 24 out of 27 samples when a reference panel is provided. In comparison, \texttt{COIL} correctly infers the number of strains in 23 samples. We notice that both methods struggle to identify strains whose relative proportions is below 5\% (Figure~\ref{fig:switchVsMisCopy}(a)). Specifically, both methods fail to detect the minor strain of 1\% in sample PG0414-C. However \texttt{COIL} is in favour of underestimating all strains with proportions no more than 5\%, whereas \texttt{DEploid} tends to over-fit the minor strain with an additional strain of the same proportion. We recommend to adjust the value of $\sigma$ for the prior to improve inference for extremely unbalanced samples in the supplementary materials.}

\textcolor{red}{The method \texttt{pfmix} infer the number of strains and proportions only based on the allele frequency imbalance within sample: It infers the strain proportions when assuming the number of strains from one to eight, then uses the Bayesian information criterion to choose the best model. As in the attempt of applying \texttt{pfmix} to the same dataset is unsuccessful, we ignore the model selection step of \texttt{pfmix}, and infer proportions directly with fixed number of strains. Similar to the comparison shown in Figure~\ref{fig:jason}, we compute the observed and expected effective number of strains of each sample, and find consistent results between \texttt{DEploid} and \texttt{pfmix}. Note that even though \texttt{DEploid} over-fits extremely unbalanced samples with an additional strain, the extra strain and its proportion has minor contribution towards the effective number of strains, which interpret the mixture as a whole.
}
\donenum{REV1.12: Figure 3: pfmix infers the number of strains.}

We also experimented with \texttt{BEAGLE} and \texttt{SHAPEIT} for deconvolving haplotypes in mixtures of two strains. \textcolor{red}{\texttt{BEAGLE} and \texttt{SHAPEIT} would implicitly assume a 50:50 distribution of alleles with its diploid assumption.} \donenum{REV1.21: BEAGLE assumption.} Both methods worked well for balanced mixtures (i.e. with proportions between 40\% and 60\%) as they mimic a diploid sample. However, as strain proportions became more unbalanced, accuracy degraded and both methods wrongly inferred heterozygous sites as homozygous, introducing a bias towards inferring the haplotypes of dominant strains. We observed that strains with a relative proportion below 20\% were always masked out by the dominant strain (Figure~\ref{fig:switchVsMisCopy}(c)).

\subsection{\textcolor{red}{Simulation from field samples}}
\textcolor{red}{
We conducted simulation studies to investigate how DEploid performs on field samples, where two scenarios of mixtures were considered: 25/75\% and 45/55\% over 8,071 sites. Twenty-two haplotypes were randomly selected from 212 Asian clonal samples, where the first twenty haplotypes were treated as candidates of the reference panel. Let $h_{21}$ and $h_{22}$ denote the genotypes for the 21st and 22nd haplotypes, the true WSAF was computed as $0.25 \times h_{21} + 0.75 \times h_{22}$ for the 25/75\% mixture, followed by adjusting from Eqn.\eqref{eqn:adj_q}. We assumed that the sequencing coverage is the same as of the 21st haplotype, and drew independent Binomial variables from the total depth with the probability of the adjusted WSAF to mimic the alternative read count, which was subtracted from the total coverage to obtain the reference allele count. DEploid correctly recovered the number of strains and proportions. As expected, we observed more switches and genotype errors in 45/55\% mixture than 25/75\% mixture, with means of 58.9 and 24.7 for switches, and 0.0184 and 0.0073 for genotype errors respectively (Figure~\ref{fig:simField}).
}\donenum{REV1.2.1: simulation}

\begin{figure}[htb]
\centering
\includegraphics[width=.45\textwidth]{{simulationPf3k_CHROM14/switches_and_genotypepanel20-seg50}.pdf}
\caption{Histograms of number of switches and genotype errors of deconvolution of 100 simulated Pf3k samples.}\label{fig:simField}
\end{figure}


\subsection{Run-time}

The complexity of our program is $\mathcal{O}(lm^2)$, where $m$ and $l$ are the number of reference strains and sites respectively. In practice, we recommend dividing samples into distinct geographical regions to perform deconvolution. \textcolor{red}{We then compute the pairwise differences between two clonal strains, and use} the ten most different \donenum{REV1.22: ``ten most different'' -- different how? Define.} local clonal strains as as reference panel. The run time for deconvolution a field sample range between 1 and 6 hours, depending on the number variants in a sample: For example, it takes $5\frac{1}{2}$ hours to process sample {\textmd QG0182-C} over 372,884 sites.  We give worked examples of deconvolving mixed infections from {\it in vitro} samples in the Supplementary Material.



%\begin{figure}[htb]
%\centering
%\includegraphics[width=.45\textwidth]{Fig4.pdf}
%\caption{Run-time and scaling.  CPU time (seconds) for deconvolving chromosomes 12, 13 and 14 of sample {\textmd PG0412-C} with reference panels I, V and VI (size 4, 10 and 16 reference haplotypes respectively). The run-time is approximately linear with respect to the number of sites and shows the expected quadratic trend against the number of reference strains.}\label{fig:runtime}
%\end{figure}


\section{Discussion}

The program \texttt{DEploid} and its analysis pipeline has been originally developed for {\it P. falciparum} studies. Nonetheless, with \textcolor{red}{some} parameter changes, \texttt{DEploid} can be used for deconvolution of any other data set with a mixture of samples from a single species, for example on data from {\it Plasmodium vivax} \citep{Pearson2016} or bacterial and viral pathogens. \textcolor{red}{However, each organism genome preserves its own unique biological signature, which reflects differently in sequence data. Variable data qualities and sequencing artefacts require different filtering steps and parameter changes. We show examples and discuss the effect of parameters in the supplementary materials. Nevertheless, the current method struggles with data both show evidence of inbreeding but low coverage. We aim to resolve these issues in near future.} \donenum{REV2.2: Application to other species (Discussion).}

There are several limitation of the current implementation, the greatest of which is the quadratic scaling with reference panel size. \textcolor{red}{Note that a typical reference panel from field samples is not perfect, and does not guarantee all the genetic `foot print' are presented. Therefore, it would be ideal to include as many reference strains as possible. However, this approach is computationally prohibitive.} \donenum{REV1.15: It is always better to deconvolve with a reference panel.} In practice, current approaches to related problems such as haplotype phasing \citep{Delaneau2012} or inference from low-coverage sequencing experiments \citep{Davis2016} typically aim to select a few candidate haplotypes (which might be a mosaic) from a reference panel.  Alternatively, the reference panel data can itself be approximated, for example through graphical structures, as in BEAGLE \citep{Browning2007}, or represented through structures that enable efficient computation \citep{Lunter2016}. \textcolor{red}{Similarly, our current implementation only processes SNPs. To reconstruct the complete haplotype, we should also consider structural variants such as insertions and deletions.}
\donenum{REV2.4: InDels and structural variants. } Such extensions will be pursued in future work.

\textcolor{red}{As technology arises, single molecule sequencing with long-read data has become available, e.g. PacBio or Oxford Nanopore Technologies. These technologies generate sequence data of lengths in kilobytes typically. Long-read data can better preserve the genetic linkage information than short-read data. Hence we expect higher quality deconvolution results when applying long-read data in principle. However, how to overcome the genotype error and artifacts in these technologies has not been tested by DEploid. As for future development, we will investigate how to adjust the model parameters to cope with different sequencing platforms.}
\donenum{REV2.3:Other sequencing technologies.}



\section*{Acknowledgements}
We thank the Pf3k consortium for valuable insights, in particular, suggestions from Roberto Amato, John O'Brien, Richard Pearson, Jerome Kelleher and Jason Wendler for providing the data of artificial samples. We thank Zam Iqbal for suggesting the name DEploid.

\section*{Funding}
Funded by the Wellcome Trust grant [100956/Z/13/Z] to GM.\\
~\\
\noindent{\em Conflict of Interest: none declared.}


\begin{thebibliography}{}

\bibitem[\protect\citeauthoryear{Arnot}{Arnot}{1998}]{Arnot1998}
Anita, D. (1998).
\newblock Unstable malaria in Sudan: the influence of the dry season: clone
  multiplicity of {\it Plasmodium falciparum} infections in individuals exposed to
  variable levels of disease transmission.
\newblock {\em Trans. R. Soc. Trop. Med. Hyg.\/}~{\em 92\/}(6), 580--585.

\bibitem[\protect\citeauthoryear{Bell, de Roode, Sim Read, and Koella}{Bell et~al.}{2006}]{Bell2006}
Bell A.~S. {\em et al.} (2006)
\newblock Within-host competition in genetically diverse malaria infection: parasite virulence and competitive success.
\newblock {\em Evolution\/}~{\em 60}(7), 1358--1371.

\bibitem[\protect\citeauthoryear{Browning and Browning}{Browning and
  Browning}{2007}]{Browning2007}
Browning, S.~R. and B.~L. Browning (2007)
\newblock Rapid and accurate haplotype phasing and missing-data inference for
  whole-genome association studies by use of localised haplotype clustering.
\newblock {\em Am. J. Hum. Genet.\/}~{\em 81\/}(5), 1084--1097.

\bibitem[\protect\citeauthoryear{Chang}{Chang et~al.}{2017}]{Chang2017}
\textcolor{red}{Change, H.~H. et al. (2017)
\newblock THE REAL McCOIL: A method for the concurrent estimation of the complexity of infection and SNP allele frequency for malaria parasites.
\newblock {\em PLoS Comput. Biol.\/}~{\em 13\/}(1), e1005348.}
\donenum{Change, H.~H. et al. (2017)}

\bibitem[\protect\citeauthoryear{Davis, Flint, Myers and Mott}{Davis}{2016}]{Davis2016}
Davies, R.~W. {\em et al}. (2016)
Rapid genotype imputation from sequence without reference panels.
\newblock{\em Nat. Genet.\/}~{\em 48\/}, 965--969.

\bibitem[\protect\citeauthoryear{Delaneau, Marchini, and Zagury}{Delaneau et~al.}{2012}]{Delaneau2012}
Delaneau, O. {\em et al}. (2012)
\newblock A linear complexity phasing method for thousands of genomes.
\newblock {\em Nat. Methods\/}~{\em 9\/}(2), 179--181.

\bibitem[\protect\citeauthoryear{de~Roode, Culleton, Bell, and Read}{de~Roode
  et~al.}{2004}]{deRoode2004}
de~Roode, J. {\em et al}. (2004)
\newblock Competitive release of drug resistance following drug treatment of mixed {\it Plasmodium chabaudi} infections.
\newblock {\em Malar. J.\/}~{\em 3\/}(33), 1--6.

\bibitem[\protect\citeauthoryear{de~Roode, Pansini, Cheesman, Helinski,
  Huijben, Wargo, Bell, Chan, Walliker, and Read}{de~Roode
  et~al.}{2005}]{deRoode2005}
de~Roode, J. {\em et al}. (2005)
\newblock Virulence and competitive ability in genetically diverse malaria infections.
\newblock {\em Proc. Natl. Acad. Sci. USA\/}~{\em 102\/}(21), 7624--7628.

\bibitem[\protect\citeauthoryear{Galinsky}{Galinsky et~al.}{2015}]{Galinsky2015}
Galinsky, K. {\em et al}. (2015)
\newblock COIL: a methodology for evaluating malarial complexity of infection using likelihood from single nucleotide polymorphism data.
\newblock {\em Malar. J.\/}~{\em14\/}(4), 1--9.

\bibitem[\protect\citeauthoryear{Harris and Nielsen}{Harris and Nielsen}{2013}]{Harris2013}
Harris K. and Nielsen R. (2013).
\newblock{Inferring Demographic History from a Spectrum of Shared Haplotype Lengths}.
\newblock{\em PLoS Genet.\/}~{\em9/}(6), e1003521.

\bibitem[\protect\citeauthoryear{Hastings and D\'Alessandro}{Hastings and
  D\'Alessandro}{2000}]{Hastings2000}
Hastings, I. and U.~D'Alessandro (2000).
\newblock Modelling a predictable disaster: the rise and spread of drug-resistant malaria.
\newblock {\em Parasitol. Today\/}~{\em 16\/}(8), 340--347.

\bibitem[\protect\citeauthoryear{Howie, Donnelly, and Marchini}{Howie
  et~al.}{2009}]{Howie2009}
Howie, B.~N. {\em et al}. (2009)
\newblock A flexible and accurate genotype imputation method for the next generation of genome-wide association studies.
\newblock {\em PLoS Genet.\/}~{\em 5\/}(6), e1000529.

\bibitem[\protect\citeauthoryear{Lawson et~al.}{Lawson et~al.}{2012}]{Lawson2012}
Lawson D.~J. {\em et al}. (2012)
\newblock{Inference of Population Structure using Dense Haplotype Data}.
\newblock{\em PLoS Genet.\/}~{\em 8\/}(1), e1002453.

\bibitem[\protect\citeauthoryear{Li and Stephens}{Li and
  Stephens}{2003}]{Li2003}
Li, N. and M.~Stephens (2003)
\newblock {Modeling linkage disequilibrium and identifying recombination
  hotspots using single-nucleotide polymorphism data}.
\newblock {\em Genetics\/}~{\em 165\/}(4), 2213--2233.

\bibitem[\protect\citeauthoryear{Lunter}{Lunter}{2016}]{Lunter2016}
Lunter, G. (2016)
\newblock {Fast haplotype matching in very large cohorts using the Li and Stephens model}.
\newblock {\em bioRxiv\/}, 10.1101/048280.

\bibitem[\protect\citeauthoryear{MalariaGEN}{MalariaGEN}{2008}]{MalariaGen2008}
MalariaGEN (2008)
\newblock A global network for investigating the genomic epidemiology of
  malaria.
\newblock {\em Nature\/}~{\em 456\/}(7223), 732--737.

\bibitem[\protect\citeauthoryear{Manske, Miotto, Campino, Auburn,
  Almagro-Garcia, Maslen, O'Brien, Djimde, Doumbo, Zongo, Ouedraogo, Michon,
  Mueller, Siba, Nzila, Borrmann, Kiara, Marsh, Jiang, Su, Amaratunga,
  Fairhurst, Socheat, Nosten, Imwong, White, Sanders, Anastasi, Alcock, Drury,
  Oyola, Quail, Turner, Ruano-Rubio, Jyothi, Amenga-Etego, Hubbart, Jeffreys,
  Rowlands, Sutherland, Roper, Mangano, Modiano, Tan, Ferdig, Amambua-Ngwa,
  Conway, Takala-Harrison, Plowe, Rayner, Rockett, Clark, Newbold, Berriman,
  MacInnis, and Kwiatkowski}{Manske et~al.}{2012}]{Manske2012}
Manske, M. {\em et al}. (2012)
\newblock Analysis of plasmodium falciparum diversity in natural infections by
  deep sequencing.
\newblock {\em Nature\/}~{\em 487\/}(7407), 375--379.

\bibitem[\protect\citeauthoryear{Mathieson and McVean}{Mathieson and McVean}{2014}]{Mathieson2014}
Mathieson I. and McVean G. (2014).
\newblock{Demography and the Age of Rare Variants}.
\newblock{\em PLoS Genet.\/}~{\em 10}(8), e1004528.

\bibitem[\protect\citeauthoryear{McKenna}{McKenna et~al.}{2010}]{McKenna2010}
\textcolor{red}{
McKenna, A. et al. (2010).
\newblock The Genome Analysis Toolkit: a MapReduce framework for analyzing next-generation DNA sequencing data.
\newblock {\em Genome Res.\/}~{\em 20}, 1297--1303.
}\donenum{McKenna, A. et al. (2010).}

\bibitem[\protect\citeauthoryear{Miles, Iqbal, Vauterin, Pearson, Campino,
  Theron, Gould, Mead, Drury, O{\textquoteright}Brien, Ruano~Rubio, MacInnis,
  Mwangi, Samarakoon, Ranford-Cartwright, Ferdig, Hayton, Su, Wellems, Rayner,
  McVean, and Kwiatkowski}{Miles et~al.}{2016}]{Miles2016}
Miles, A. {\em et al}. (2015)
\newblock Indels, structural variation, and recombination drive genomic diversity in {\it Plasmodium falciparum}.
\newblock {\em Genome Res.\/}~{\em26\/}, 1288--1299.

\bibitem[\protect\citeauthoryear{Pearson, Amato, Auburn, Miotto,
  Almagro-Garcia, Amaratunga, Suon, Mao, Noviyanti, Trimarsanto, Marfurt,
  Anstey, William, Boni, Dolecek, Tran, White, Michon, Siba, Tavul, Harrison,
  Barry, Mueller, Ferreira, Karunaweera, Randrianarivelojosia, Gao, Hubbart,
  Hart, Jeffery, Drury, Mead, Kekre, Campino, Manske, Cornelius, MacInnis,
  Rockett, Miles, Rayner, Fairhurst, Nosten, Price, and Kwiatkowski}{Pearson
  et~al.}{2016}]{Pearson2016}
Pearson, R.~D. {\em et al}. (2016)
\newblock {Genomic analysis of local variation and recent evolution in {\it Plasmodium vivax}}.
\newblock {\em Nat. Genet.\/}~{\em 48}, 959--964.

\bibitem[\protect\citeauthoryear{O\'Connell, Gurdasani and Delaneau}{O'Connell et~al.}{2016}]{Oconnell2014}
O'Connell J., {\em et al}. (2014)
\newblock {A General Approach for Haplotype Phasing across the Full Spectrum of Relatedness}.
\newblock {\em PLoS Genet.\/}~{\em 10\/}(4), e1004234.

\bibitem[\protect\citeauthoryear{Pf3k}{Pf3k}{2016}]{Pf3k2016}
The Pf3k Project: pilot data release 5 (2016)
\newblock {www.malariagen.net/data/pf3k-5} [accessed 1 June 2016]

\bibitem[\protect\citeauthoryear{O'Brien}{O'Brien et~al.}{2016}]{Jack2016}
O'Brien D,J. {\em et al}. (2016)
\newblock Inferring Strain Mixture within Clinical {\em Plasmodium falciparum} Isolates from Genomic Sequence Data.
\newblock {\em PLoS Comput. Biol.\/}~{\em 12\/}(6): e1004824.

\bibitem[\protect\citeauthoryear{Sabetil,Reich,Higgins, Levine,Richter}{Sabetil et~al.}{2002}]{Sabeti2002}
Sabeti1. P.~C. {\em et al}. (2002)
\newblock {Detecting recent positive selection in the human genome from haplotype structure}.
\newblock {\em Nature\/}~{\em 419\/}, 832--837.

\bibitem[\protect\citeauthoryear{Wendler}{Wendler}{2015}]{Wendler2015}
Wendler, J. (2015)
\newblock {\em Accessing complex genomic variation in} {P}lasmodium falciparum {\em natural infection}.
\newblock {Ph.\ D. thesis, University of Oxford.}

\bibitem[\protect\citeauthoryear{WHO}{WHO}{2016}]{WHO2016}
WHO. (2016)
\newblock {World Malaria Report 2015}.
\newblock {\em World Health Organization\/}.
\end{thebibliography}
\end{document}
