\input{supplementReset.tex}

%\begin{center}
%\textbf{\large Supplemental Materials of DEploid}
%\end{center}

\section{DEploid examples} \label{sup:sec:deploid}

Our program {\it DEploid } is freely available, including C++ source code at \url{https://github.com/mcveanlab/DEploid} under the conditions of the GPLv3 license. A detailed document can be found at \url{http://deploid.readthedocs.io/en/latest/}. An R version is available at \url{https://github.com/mcveanlab/DEploid-r}.

\subsection{Data exploration example}
As a demonstration, we show some examples of mixed {\em P. falciparum} genome deconvolution using one of the reference panels described in the main article (reference panel V). All data used in this chapter is avaiable at \url{https://github.com/shajoezhu/DEploid-Supplementary-Materials}. First, we use an {\tt R} script to plot and explore the data of a mixed sample. For instance, sample {\textmd PG0395-C} is a mixture of strains HB3, 7G8 and Dd2 with equal proportions, we run the following command
\linespread{1}
\begin{lstlisting}
R --slave "--args -vcf PG0395-C.wg.vcf.gz
    -plaf labStrains_PLAF.txt
    -exclude labStrainsExclude.txt
    -o PG0395-C" < ~/DEploid/utilities/dataExplore.r
\end{lstlisting}
\linespread{1.5}
where ``{\tt -vcf PG0395-C.wg.vcf.gz}'' defines the input VCF file. Notice that, in this example, we assume all variant sites are single nucleotide polymorphism, and every site is tagged with {\tt PASS} at the {\tt QUAL} column. The read counts of the reference and alternative alleles must be presented in the {\tt AD} field at all sites; ``{\tt -plaf labStrains\_PLAF.txt}'' contains the population allele frequencies calculated from total read counts (see main article secion 2.1 Notations). Note that we use option ``{\tt -exclude labStrainsExclude.txt}'' to skip deconvoluting sites of which the alternative allele counts are zeros.

\begin{figure}[ht]
\centering
\includegraphics[width=.8\textwidth]{{supplementDEploidExample/PG0395-CaltVsRefAndWSAFvsPLAF}.png}
\caption{Data exploration of sample {\textmd PG0395-C}.}\label{fig:0395}
\end{figure}

From left to right, Fig.~\ref{fig:0395} shows:
\begin{enumerate}
\item Alternative read counts vs reference read counts. The number of clusters or blobs in the figure can provide some intuition on the number of mixed strains, see Fig~\ref{fig:0396}~(a). However, this can be misleading, as the number of clusters does not always reflect the true number of mixed strains. In the example shown in Fig.~\ref{fig:0395}, the alternative vs reference counts figure only show two clusters, but the true number of mixture is three.

\item Histogram of the allele frequencies within sample. Similar to the alternative vs reference counts figure, the number of WSAF modes can be used as an indication of the number of strains and proportions within a field isolate (Fig~\ref{fig:0396}~(a)). Again, this is not always true, Fig.~\ref{fig:0395} shows that two modes of WSAF at approximately 0.33 and 0.66, but the true number of mixed strains is three, and the proportions are 1/3, 1/3 and 1/3.

\item Allele frequencies at the population level (PLAF) vs allele frequencies within sample (WSAF). The PLAF is calculated from total read counts (see main article secion 2.1 Notations), whereas the WSAF is calculated by using allele counts.
\end{enumerate}

\subsection{Deconvoluting sample {\textmd PG0396-C}}
The following example shows a specific {\textmd DEploid} command to deconvolute the mixed sample {\textmd PG0396-C}:
\linespread{1}
\begin{lstlisting}
dEploid -vcf PG0396-C.wg.vcf.gz \
    -plaf labStrains_PLAF.txt \
    -exclude labStrainsExclude.txt \
    -panel labStrainsPanelFinal.txt \
    -seed 5 \
    -nSample 250 \
    -rate 8 \
    -burn 0.67 \
    -k 3 \
    -o PG0396-C_seed5k3 \
    -exportPostProb
\end{lstlisting}
\linespread{1.5}
where flags ``{\tt -vcf}'', ``{\tt -plaf}'' and ``{\tt -exclude}'' are used in the same manner as in the previous example; ``{\tt -panel labStrainsPanelFinal.txt}'' specifies a text file including haplotypes of lab strains 3D7, HB3, 7G8 and Dd2; options ``{\tt -nSample}'', ``{\tt -rate}'' and ``{\tt -burn}'' specify the total number of MCMC samples to take, the sampling rate and the burning rate of the MCMC chain respectively. For detailed documentation, please see \url{http://deploid.readthedocs.io/en/latest/input.html}.

We use a utility {\tt R} script to plot and interpret the output produced by DEploid. The following command is used to generate Figures~\ref{fig:0396} (a) -- (e).
\linespread{1}
\begin{lstlisting}
R --slave "--args -vcf PG0396-C.wg.vcf.gz
    -plaf labStrains_PLAF.txt
    -exclude labStrainsExclude.txt
    -dEprefix PG0396-C_seed5k3
    -o PG0396-C_seed5k3" < ~/DEploid/utilities/interpretDEploid.r
\end{lstlisting}
\linespread{1.5}

\begin{figure}[ht]
\centering
\subfloat[][]{
\includegraphics[width=.47\textwidth]{{supplementDEploidExample/PG0396-C_seed5k3.interpretDEploidFigure.1}.png}
}%\label{fig:pg0396.interpret1}
\subfloat[][]{
\includegraphics[width=.53\textwidth]{{supplementDEploidExample/PG0396-C_seed5k3.interpretDEploidFigure.2}.png}
}\\
%\caption{}
%\end{figure}

%\begin{figure}[ht]
%\ContinuedFloat
\subfloat[][]{
\includegraphics[width=.32\textwidth]{{supplementDEploidExample/PG0396-C_seed5k3.single0}.png}
}
\subfloat[][]{
\includegraphics[width=.32\textwidth]{{supplementDEploidExample/PG0396-C_seed5k3.single1}.png}
}
\subfloat[][]{
\includegraphics[width=.32\textwidth]{{supplementDEploidExample/PG0396-C_seed5k3.single2}.png}
}\\
\caption{Sample {\textmd PG0396-C} deconvolution with reference panel V.}\label{fig:0396}
\end{figure}

Here, we briefly describe each panel in Fig.~\ref{fig:0396}:
\begin{itemize}

\item (a) Diagnostic panels from the {\textmd DEploid} output. The top three panels recap the data exploration process, with an enhanced PLAF vs WASF plot: red dots show observed WSAF, which is calculated by read counts; blue does show the expected WSAF inferred from our model (see Eqn.~(3) in main article). The next three plots from left to right show:
\begin{enumerate}
\item[4.] MCMC samples for the strain proportions, with the fraction of each color indicating the proportion of a different strain at each MCMC sample. The three colored blocks suggest that there are three strains within sample {\textmd PG0396-C}, with proportions approximately 1/4, 1/2 and 1/4.
\item[5.] Expected WSAF vs observed WSAF. We use the correlation between the observed and expected WSAF as a sanity check for our model. A low correlation suggests poor fitting.
\item[6.] Log likelihood of the MCMC chain. This figure indicates whether the MCMC has converged. The colored dots mark the likelihoods of the model when specific MCMC steps are used: updating the strain porportions, painting a single haplotype and painting a pair of haplotypes are marked in green red and blue respectively.

\end{enumerate}
\item (b) Expected WSAF (blue) and observed WSAF (red) across the genome. This figure highlights the genome diversity within the mixed sample across the genome.
\item (c) (d) and (e) show the posterior painting probabilities for the deconvoluted strains when using the reference panel V. In each figure, 3D7, HB3, 7G8 and Dd2 are represented by colors red, light orange, yellow and dark orange respectively. Each panel represents the posterior probability of a chromosome. Chromosomes 1 -- 14 are ordered from left to right, then top to bottom.
\end{itemize}

In this example, the second inferred haplotype (Fig.~\ref{fig:0396}~(d)) represents the 7G8 strain, which has relative proportion of 1/2. The remaining two strains have approximately equal proportions, which increase the difficulty to deconvolute them. In practice, we find more switching errors when deconvoluting samples containing strains at similar proportions, for example, Fig.~\ref{fig:0396}~(c) and (e) both show high probabilities of copying from strains HB3 and Dd2. More specificly, Fig.~\ref{fig:0396}~(c) panel 1 shows almost every position of inferred haplotype chromosome 1 is copying from strain HB3; but chromosome 2 is copying from strain Dd2; chromosome 3 is partly copying from strains HB3 and Dd2, with presence of one switching error. These observations suggest that our program DEploid can characterize the main genome diversities within a mixed samples, yet there is still room to improve to overcome the switching errors.

\subsection{Deconvoluting sample {\textmd PG0415-C}}
As metioned in the main article, we assume that there are more strains than necessary, and only keep strains with inferred relative proportions greater than 0.01 in practice. In the following example, we show the deconvolution of a clonal sample ({\textmd PG0415-C}) when starting from three strains.
\linespread{1}
\begin{lstlisting}
dEploid -vcf PG0415-C.wg.vcf.gz \
    -panel labStrainsPanelFinal.csv \
    -plaf labStrains_PLAF.txt \
    -exclude labStrainsExclude.txt \
    -seed 2 \
    -nSample 250 \
    -rate 8 \
    -burn 0.67 \
    -k 3 \
    -o PG0415-C_seed2k3 \
    -exportPostProb
\end{lstlisting}

\linespread{1.5}
\begin{figure}[th]
\subfloat[][]{
\includegraphics[width=.47\textwidth]{{supplementDEploidExample/PG0415-C_seed2k3.interpretDEploidFigure.1}.png}
}
\subfloat[][]{
\includegraphics[width=.53\textwidth]{{supplementDEploidExample/PG0415-C_seed2k3.interpretDEploidFigure.2}.png}
}\\
\subfloat[][]{
\includegraphics[width=.32\textwidth]{{supplementDEploidExample/PG0415-C_seed2k3.single0}.png}
}\label{fig:pg0415.postprob_a}
\subfloat[][]{
\includegraphics[width=.32\textwidth]{{supplementDEploidExample/PG0415-C_seed2k3.single1}.png}
}\label{fig:pg0415.postprob_b}
\subfloat[][]{
\includegraphics[width=.32\textwidth]{{supplementDEploidExample/PG0415-C_seed2k3.single2}.png}
}\label{fig:pg0415.postprob_c}\\
\caption{Sample {\textmd PG0415-C} deconvolution with reference panel V.}\label{fig:0415}
\end{figure}
Fig.~\ref{fig:0415}~(a) and (e) suggests that our program successfully dropped out one strain, reduced the number of strains to two. However, our model overfits this clonal sample as a mixture of two strains. This is caused by a number of problematic sites with both high number of alternative and reference allele counts, resulting in high leverage in our model. When we paint the haplotypes with the reference panels, Fig.~\ref{fig:0415} (c) and (d) suggest that both haplotypes are in fact copying from the same strain (7G8).
